// @radixdlt/babylon-gateway-api-sdk/dist/babylon-gateway-api-sdk.mjs
var Ee = Object.defineProperty;
var qe = (e, t, n) => t in e ? Ee(e, t, { enumerable: !0, configurable: !0, writable: !0, value: n }) : e[t] = n;
var R = (e, t, n) => (qe(e, typeof t != "symbol" ? t + "" : t, n), n);
const De = "https://betanet.radixdlt.com".replace(/\/+$/, "");
class Ae {
  constructor(t = {}) {
    this.configuration = t;
  }
  set config(t) {
    this.configuration = t;
  }
  get basePath() {
    return this.configuration.basePath != null ? this.configuration.basePath : De;
  }
  get fetchApi() {
    return this.configuration.fetchApi;
  }
  get middleware() {
    return this.configuration.middleware || [];
  }
  get queryParamsStringify() {
    return this.configuration.queryParamsStringify || V;
  }
  get username() {
    return this.configuration.username;
  }
  get password() {
    return this.configuration.password;
  }
  get apiKey() {
    const t = this.configuration.apiKey;
    if (t)
      return typeof t == "function" ? t : () => t;
  }
  get accessToken() {
    const t = this.configuration.accessToken;
    if (t)
      return typeof t == "function" ? t : async () => t;
  }
  get headers() {
    return this.configuration.headers;
  }
  get credentials() {
    return this.configuration.credentials;
  }
}
const ve = new Ae();
class S {
  constructor(t = ve) {
    R(this, "middleware");
    R(this, "fetchApi", async (t, n) => {
      let r = { url: t, init: n };
      for (const o of this.middleware)
        o.pre && (r = await o.pre({
          fetch: this.fetchApi,
          ...r
        }) || r);
      let s;
      try {
        s = await (this.configuration.fetchApi || fetch)(r.url, r.init);
      } catch (o) {
        for (const a of this.middleware)
          a.onError && (s = await a.onError({
            fetch: this.fetchApi,
            url: r.url,
            init: r.init,
            error: o,
            response: s ? s.clone() : void 0
          }) || s);
        if (s === void 0)
          throw o instanceof Error ? new Pe(o, "The request failed and the interceptors did not return an alternative response") : o;
      }
      for (const o of this.middleware)
        o.post && (s = await o.post({
          fetch: this.fetchApi,
          url: r.url,
          init: r.init,
          response: s.clone()
        }) || s);
      return s;
    });
    this.configuration = t, this.middleware = t.middleware;
  }
  withMiddleware(...t) {
    const n = this.clone();
    return n.middleware = n.middleware.concat(...t), n;
  }
  withPreMiddleware(...t) {
    const n = t.map((r) => ({ pre: r }));
    return this.withMiddleware(...n);
  }
  withPostMiddleware(...t) {
    const n = t.map((r) => ({ post: r }));
    return this.withMiddleware(...n);
  }
  async request(t, n) {
    const { url: r, init: s } = await this.createFetchParams(t, n), o = await this.fetchApi(r, s);
    if (o.status >= 200 && o.status < 300)
      return o;
    throw new ke(o, "Response returned an error code");
  }
  async createFetchParams(t, n) {
    let r = this.configuration.basePath + t.path;
    t.query !== void 0 && Object.keys(t.query).length !== 0 && (r += "?" + this.configuration.queryParamsStringify(t.query));
    const s = Object.assign({}, this.configuration.headers, t.headers);
    Object.keys(s).forEach((M) => s[M] === void 0 ? delete s[M] : {});
    const o = typeof n == "function" ? n : async () => n, a = {
      method: t.method,
      headers: s,
      body: t.body,
      credentials: this.configuration.credentials
    }, O = {
      ...a,
      ...await o({
        init: a,
        context: t
      })
    }, Ie = {
      ...O,
      body: Ce(O.body) || O.body instanceof URLSearchParams || xe(O.body) ? O.body : JSON.stringify(O.body)
    };
    return { url: r, init: Ie };
  }
  clone() {
    const t = this.constructor, n = new t(this.configuration);
    return n.middleware = this.middleware.slice(), n;
  }
}
function xe(e) {
  return typeof Blob < "u" && e instanceof Blob;
}
function Ce(e) {
  return typeof FormData < "u" && e instanceof FormData;
}
class ke extends Error {
  constructor(n, r) {
    super(r);
    R(this, "name", "ResponseError");
    this.response = n;
  }
}
class Pe extends Error {
  constructor(n, r) {
    super(r);
    R(this, "name", "FetchError");
    this.cause = n;
  }
}
class l extends Error {
  constructor(n, r) {
    super(r);
    R(this, "name", "RequiredError");
    this.field = n;
  }
}
const gr = {
  csv: ",",
  ssv: " ",
  tsv: "	",
  pipes: "|"
};
function i(e, t) {
  const n = e[t];
  return n != null;
}
function V(e, t = "") {
  return Object.keys(e).map((n) => L(n, e[n], t)).filter((n) => n.length > 0).join("&");
}
function L(e, t, n = "") {
  const r = n + (n.length ? `[${e}]` : e);
  if (t instanceof Array) {
    const s = t.map((o) => encodeURIComponent(String(o))).join(`&${encodeURIComponent(r)}=`);
    return `${encodeURIComponent(r)}=${s}`;
  }
  if (t instanceof Set) {
    const s = Array.from(t);
    return L(e, s, n);
  }
  return t instanceof Date ? `${encodeURIComponent(r)}=${encodeURIComponent(t.toISOString())}` : t instanceof Object ? V(t, r) : `${encodeURIComponent(r)}=${encodeURIComponent(String(t))}`;
}
function Or(e, t) {
  return Object.keys(e).reduce(
    (n, r) => ({ ...n, [r]: t(e[r]) }),
    {}
  );
}
function Rr(e) {
  for (const t of e)
    if (t.contentType === "multipart/form-data")
      return !0;
  return !1;
}
class u {
  constructor(t, n = (r) => r) {
    this.raw = t, this.transformer = n;
  }
  async value() {
    return this.transformer(await this.raw.json());
  }
}
class Sr {
  constructor(t) {
    this.raw = t;
  }
  async value() {
  }
}
class Nr {
  constructor(t) {
    this.raw = t;
  }
  async value() {
    return await this.raw.blob();
  }
}
class hr {
  constructor(t) {
    this.raw = t;
  }
  async value() {
    return await this.raw.text();
  }
}
function Fr(e) {
  return !0;
}
function f(e) {
  return Me(e);
}
function Me(e, t) {
  return e == null ? e : {
    state_version: i(e, "state_version") ? e.state_version : void 0,
    timestamp: i(e, "timestamp") ? e.timestamp === null ? null : new Date(e.timestamp) : void 0,
    epoch: i(e, "epoch") ? e.epoch : void 0,
    round: i(e, "round") ? e.round : void 0
  };
}
function _(e) {
  if (e !== void 0)
    return e === null ? null : {
      state_version: e.state_version,
      timestamp: e.timestamp === void 0 ? void 0 : e.timestamp === null ? null : e.timestamp.toISOString(),
      epoch: e.epoch,
      round: e.round
    };
}
function br(e) {
  return !0;
}
function Tr(e) {
  return Ve(e);
}
function Ve(e, t) {
  return e == null ? e : {
    at_ledger_state: i(e, "at_ledger_state") ? f(e.at_ledger_state) : void 0
  };
}
function wr(e) {
  if (e !== void 0)
    return e === null ? null : {
      at_ledger_state: _(e.at_ledger_state)
    };
}
function Jr(e) {
  let t = !0;
  return t = t && "value" in e, t;
}
function D(e) {
  return Le(e);
}
function Le(e, t) {
  return e == null ? e : {
    value: e.value,
    address: i(e, "address") ? e.address : void 0
  };
}
function A(e) {
  if (e !== void 0)
    return e === null ? null : {
      value: e.value,
      address: e.address
    };
}
const Ir = {
  Unknown: "unknown",
  CommittedSuccess: "committed_success",
  CommittedFailure: "committed_failure",
  Pending: "pending",
  Rejected: "rejected"
};
function N(e) {
  return Ke(e);
}
function Ke(e, t) {
  return e;
}
function Er(e) {
  return e;
}
function qr(e) {
  let t = !0;
  return t = t && "state_version" in e, t = t && "transaction_status" in e, t = t && "payload_hash_hex" in e, t = t && "intent_hash_hex" in e, t;
}
function v(e) {
  return Ge(e);
}
function Ge(e, t) {
  return e == null ? e : {
    state_version: e.state_version,
    transaction_status: N(e.transaction_status),
    payload_hash_hex: e.payload_hash_hex,
    intent_hash_hex: e.intent_hash_hex,
    fee_paid: i(e, "fee_paid") ? D(e.fee_paid) : void 0,
    confirmed_at: i(e, "confirmed_at") ? e.confirmed_at === null ? null : new Date(e.confirmed_at) : void 0,
    error_message: i(e, "error_message") ? e.error_message : void 0
  };
}
function x(e) {
  if (e !== void 0)
    return e === null ? null : {
      state_version: e.state_version,
      transaction_status: e.transaction_status,
      payload_hash_hex: e.payload_hash_hex,
      intent_hash_hex: e.intent_hash_hex,
      fee_paid: A(e.fee_paid),
      confirmed_at: e.confirmed_at === void 0 ? void 0 : e.confirmed_at === null ? null : e.confirmed_at.toISOString(),
      error_message: e.error_message
    };
}
function Dr(e) {
  let t = !0;
  return t = t && "address" in e, t;
}
function Ar(e) {
  return Ue(e);
}
function Ue(e, t) {
  return e == null ? e : {
    at_ledger_state: i(e, "at_ledger_state") ? f(e.at_ledger_state) : void 0,
    address: e.address
  };
}
function $e(e) {
  if (e !== void 0)
    return e === null ? null : {
      at_ledger_state: _(e.at_ledger_state),
      address: e.address
    };
}
const vr = {
  FungibleResource: "fungible_resource",
  NonFungibleResource: "non_fungible_resource",
  Package: "package",
  Component: "component"
};
function h(e) {
  return Be(e);
}
function Be(e, t) {
  return e;
}
function xr(e) {
  return e;
}
function Cr(e) {
  let t = !0;
  return t = t && "discriminator" in e, t = t && "package_address" in e, t = t && "blueprint_name" in e, t = t && "state" in e, t = t && "access_rules_chain" in e, t;
}
function kr(e) {
  return K(e);
}
function K(e, t) {
  return e == null ? e : {
    discriminator: h(e.discriminator),
    package_address: e.package_address,
    blueprint_name: e.blueprint_name,
    state: e.state,
    access_rules_chain: e.access_rules_chain
  };
}
function ze(e) {
  if (e !== void 0)
    return e === null ? null : {
      discriminator: e.discriminator,
      package_address: e.package_address,
      blueprint_name: e.blueprint_name,
      state: e.state,
      access_rules_chain: e.access_rules_chain
    };
}
function Pr(e) {
  let t = !0;
  return t = t && "discriminator" in e, t = t && "access_rules_chain" in e, t = t && "vault_access_rules_chain" in e, t = t && "divisibility" in e, t = t && "total_supply" in e, t = t && "total_minted" in e, t = t && "total_burnt" in e, t;
}
function Mr(e) {
  return G(e);
}
function G(e, t) {
  return e == null ? e : {
    discriminator: h(e.discriminator),
    access_rules_chain: e.access_rules_chain,
    vault_access_rules_chain: e.vault_access_rules_chain,
    divisibility: e.divisibility,
    total_supply: e.total_supply,
    total_minted: e.total_minted,
    total_burnt: e.total_burnt
  };
}
function He(e) {
  if (e !== void 0)
    return e === null ? null : {
      discriminator: e.discriminator,
      access_rules_chain: e.access_rules_chain,
      vault_access_rules_chain: e.vault_access_rules_chain,
      divisibility: e.divisibility,
      total_supply: e.total_supply,
      total_minted: e.total_minted,
      total_burnt: e.total_burnt
    };
}
const Vr = {
  String: "string",
  Number: "number",
  Bytes: "bytes",
  Uuid: "uuid"
};
function C(e) {
  return We(e);
}
function We(e, t) {
  return e;
}
function Lr(e) {
  return e;
}
function Kr(e) {
  let t = !0;
  return t = t && "discriminator" in e, t = t && "access_rules_chain" in e, t = t && "vault_access_rules_chain" in e, t = t && "non_fungible_id_type" in e, t;
}
function Gr(e) {
  return U(e);
}
function U(e, t) {
  return e == null ? e : {
    discriminator: h(e.discriminator),
    access_rules_chain: e.access_rules_chain,
    vault_access_rules_chain: e.vault_access_rules_chain,
    non_fungible_id_type: C(e.non_fungible_id_type)
  };
}
function Qe(e) {
  if (e !== void 0)
    return e === null ? null : {
      discriminator: e.discriminator,
      access_rules_chain: e.access_rules_chain,
      vault_access_rules_chain: e.vault_access_rules_chain,
      non_fungible_id_type: e.non_fungible_id_type
    };
}
function Ur(e) {
  let t = !0;
  return t = t && "discriminator" in e, t = t && "code_hex" in e, t;
}
function $r(e) {
  return $(e);
}
function $(e, t) {
  return e == null ? e : {
    discriminator: h(e.discriminator),
    code_hex: e.code_hex
  };
}
function Xe(e) {
  if (e !== void 0)
    return e === null ? null : {
      discriminator: e.discriminator,
      code_hex: e.code_hex
    };
}
function B(e) {
  return Ye(e);
}
function Ye(e, t) {
  if (e == null)
    return e;
  switch (e.discriminator) {
    case "component":
      return { ...K(e), discriminator: "component" };
    case "fungible_resource":
      return { ...G(e), discriminator: "fungible_resource" };
    case "non_fungible_resource":
      return { ...U(e), discriminator: "non_fungible_resource" };
    case "package":
      return { ...$(e), discriminator: "package" };
    default:
      throw new Error(`No variant of EntityDetailsResponseDetails exists with 'discriminator=${e.discriminator}'`);
  }
}
function z(e) {
  if (e !== void 0) {
    if (e === null)
      return null;
    switch (e.discriminator) {
      case "component":
        return ze(e);
      case "fungible_resource":
        return He(e);
      case "non_fungible_resource":
        return Qe(e);
      case "package":
        return Xe(e);
      default:
        throw new Error(`No variant of EntityDetailsResponseDetails exists with 'discriminator=${e.discriminator}'`);
    }
  }
}
function Br(e) {
  let t = !0;
  return t = t && "key" in e, t = t && "value" in e, t;
}
function H(e) {
  return Ze(e);
}
function Ze(e, t) {
  return e == null ? e : {
    key: e.key,
    value: e.value
  };
}
function W(e) {
  if (e !== void 0)
    return e === null ? null : {
      key: e.key,
      value: e.value
    };
}
function zr(e) {
  let t = !0;
  return t = t && "items" in e, t;
}
function m(e) {
  return je(e);
}
function je(e, t) {
  return e == null ? e : {
    total_count: i(e, "total_count") ? e.total_count : void 0,
    previous_cursor: i(e, "previous_cursor") ? e.previous_cursor : void 0,
    next_cursor: i(e, "next_cursor") ? e.next_cursor : void 0,
    items: e.items.map(H)
  };
}
function p(e) {
  if (e !== void 0)
    return e === null ? null : {
      total_count: e.total_count,
      previous_cursor: e.previous_cursor,
      next_cursor: e.next_cursor,
      items: e.items.map(W)
    };
}
function Hr(e) {
  let t = !0;
  return t = t && "network" in e, t = t && "state_version" in e, t = t && "proposer_round_timestamp" in e, t = t && "epoch" in e, t = t && "round" in e, t;
}
function d(e) {
  return et(e);
}
function et(e, t) {
  return e == null ? e : {
    network: e.network,
    state_version: e.state_version,
    proposer_round_timestamp: e.proposer_round_timestamp,
    epoch: e.epoch,
    round: e.round
  };
}
function c(e) {
  if (e !== void 0)
    return e === null ? null : {
      network: e.network,
      state_version: e.state_version,
      proposer_round_timestamp: e.proposer_round_timestamp,
      epoch: e.epoch,
      round: e.round
    };
}
function Wr(e) {
  let t = !0;
  return t = t && "ledger_state" in e, t = t && "address" in e, t = t && "metadata" in e, t;
}
function tt(e) {
  return nt(e);
}
function nt(e, t) {
  return e == null ? e : {
    ledger_state: d(e.ledger_state),
    address: e.address,
    metadata: m(e.metadata),
    details: i(e, "details") ? B(e.details) : void 0
  };
}
function Qr(e) {
  if (e !== void 0)
    return e === null ? null : {
      ledger_state: c(e.ledger_state),
      address: e.address,
      metadata: p(e.metadata),
      details: z(e.details)
    };
}
function Xr(e) {
  let t = !0;
  return t = t && "address" in e, t = t && "metadata" in e, t;
}
function Yr(e) {
  return rt(e);
}
function rt(e, t) {
  return e == null ? e : {
    address: e.address,
    metadata: m(e.metadata),
    details: i(e, "details") ? B(e.details) : void 0
  };
}
function Zr(e) {
  if (e !== void 0)
    return e === null ? null : {
      address: e.address,
      metadata: p(e.metadata),
      details: z(e.details)
    };
}
function jr(e) {
  let t = !0;
  return t = t && "address" in e, t;
}
function ei(e) {
  return it(e);
}
function it(e, t) {
  return e == null ? e : {
    at_ledger_state: i(e, "at_ledger_state") ? f(e.at_ledger_state) : void 0,
    address: e.address,
    cursor: i(e, "cursor") ? e.cursor : void 0,
    limit: i(e, "limit") ? e.limit : void 0
  };
}
function st(e) {
  if (e !== void 0)
    return e === null ? null : {
      at_ledger_state: _(e.at_ledger_state),
      address: e.address,
      cursor: e.cursor,
      limit: e.limit
    };
}
function ti(e) {
  let t = !0;
  return t = t && "address" in e, t = t && "amount" in e, t;
}
function Q(e) {
  return ot(e);
}
function ot(e, t) {
  return e == null ? e : {
    address: e.address,
    amount: D(e.amount)
  };
}
function X(e) {
  if (e !== void 0)
    return e === null ? null : {
      address: e.address,
      amount: A(e.amount)
    };
}
function ni(e) {
  let t = !0;
  return t = t && "items" in e, t;
}
function F(e) {
  return at(e);
}
function at(e, t) {
  return e == null ? e : {
    total_count: i(e, "total_count") ? e.total_count : void 0,
    previous_cursor: i(e, "previous_cursor") ? e.previous_cursor : void 0,
    next_cursor: i(e, "next_cursor") ? e.next_cursor : void 0,
    items: e.items.map(Q)
  };
}
function b(e) {
  if (e !== void 0)
    return e === null ? null : {
      total_count: e.total_count,
      previous_cursor: e.previous_cursor,
      next_cursor: e.next_cursor,
      items: e.items.map(X)
    };
}
function ri(e) {
  let t = !0;
  return t = t && "ledger_state" in e, t = t && "address" in e, t = t && "fungibles" in e, t;
}
function ut(e) {
  return dt(e);
}
function dt(e, t) {
  return e == null ? e : {
    ledger_state: d(e.ledger_state),
    address: e.address,
    fungibles: F(e.fungibles)
  };
}
function ii(e) {
  if (e !== void 0)
    return e === null ? null : {
      ledger_state: c(e.ledger_state),
      address: e.address,
      fungibles: b(e.fungibles)
    };
}
function si(e) {
  let t = !0;
  return t = t && "address" in e, t = t && "fungibles" in e, t;
}
function oi(e) {
  return ct(e);
}
function ct(e, t) {
  return e == null ? e : {
    address: e.address,
    fungibles: F(e.fungibles)
  };
}
function ai(e) {
  if (e !== void 0)
    return e === null ? null : {
      address: e.address,
      fungibles: b(e.fungibles)
    };
}
function ui(e) {
  let t = !0;
  return t = t && "items" in e, t;
}
function di(e) {
  return lt(e);
}
function lt(e, t) {
  return e == null ? e : {
    items: e.items.map(H)
  };
}
function ci(e) {
  if (e !== void 0)
    return e === null ? null : {
      items: e.items.map(W)
    };
}
function li(e) {
  let t = !0;
  return t = t && "address" in e, t;
}
function fi(e) {
  return ft(e);
}
function ft(e, t) {
  return e == null ? e : {
    at_ledger_state: i(e, "at_ledger_state") ? f(e.at_ledger_state) : void 0,
    address: e.address,
    cursor: i(e, "cursor") ? e.cursor : void 0,
    limit: i(e, "limit") ? e.limit : void 0
  };
}
function _t(e) {
  if (e !== void 0)
    return e === null ? null : {
      at_ledger_state: _(e.at_ledger_state),
      address: e.address,
      cursor: e.cursor,
      limit: e.limit
    };
}
function _i(e) {
  let t = !0;
  return t = t && "address" in e, t;
}
function mi(e) {
  return mt(e);
}
function mt(e, t) {
  return e == null ? e : {
    address: e.address,
    cursor: i(e, "cursor") ? e.cursor : void 0,
    limit: i(e, "limit") ? e.limit : void 0
  };
}
function pi(e) {
  if (e !== void 0)
    return e === null ? null : {
      address: e.address,
      cursor: e.cursor,
      limit: e.limit
    };
}
function yi(e) {
  let t = !0;
  return t = t && "ledger_state" in e, t = t && "address" in e, t = t && "metadata" in e, t;
}
function pt(e) {
  return yt(e);
}
function yt(e, t) {
  return e == null ? e : {
    ledger_state: d(e.ledger_state),
    address: e.address,
    metadata: m(e.metadata)
  };
}
function gi(e) {
  if (e !== void 0)
    return e === null ? null : {
      ledger_state: c(e.ledger_state),
      address: e.address,
      metadata: p(e.metadata)
    };
}
function Oi(e) {
  let t = !0;
  return t = t && "address" in e, t = t && "metadata" in e, t;
}
function Ri(e) {
  return gt(e);
}
function gt(e, t) {
  return e == null ? e : {
    address: e.address,
    metadata: m(e.metadata)
  };
}
function Si(e) {
  if (e !== void 0)
    return e === null ? null : {
      address: e.address,
      metadata: p(e.metadata)
    };
}
function Ni(e) {
  let t = !0;
  return t = t && "address" in e, t;
}
function hi(e) {
  return Ot(e);
}
function Ot(e, t) {
  return e == null ? e : {
    at_ledger_state: i(e, "at_ledger_state") ? f(e.at_ledger_state) : void 0,
    address: e.address,
    resource_address: i(e, "resource_address") ? e.resource_address : void 0,
    cursor: i(e, "cursor") ? e.cursor : void 0,
    limit: i(e, "limit") ? e.limit : void 0
  };
}
function Rt(e) {
  if (e !== void 0)
    return e === null ? null : {
      at_ledger_state: _(e.at_ledger_state),
      address: e.address,
      resource_address: e.resource_address,
      cursor: e.cursor,
      limit: e.limit
    };
}
function Fi(e) {
  let t = !0;
  return t = t && "address" in e, t;
}
function bi(e) {
  return St(e);
}
function St(e, t) {
  return e == null ? e : {
    address: e.address,
    resource_address: i(e, "resource_address") ? e.resource_address : void 0,
    cursor: i(e, "cursor") ? e.cursor : void 0,
    limit: i(e, "limit") ? e.limit : void 0
  };
}
function Ti(e) {
  if (e !== void 0)
    return e === null ? null : {
      address: e.address,
      resource_address: e.resource_address,
      cursor: e.cursor,
      limit: e.limit
    };
}
function wi(e) {
  let t = !0;
  return t = t && "non_fungible_id" in e, t;
}
function Y(e) {
  return Nt(e);
}
function Nt(e, t) {
  return e == null ? e : {
    non_fungible_id: e.non_fungible_id
  };
}
function Z(e) {
  if (e !== void 0)
    return e === null ? null : {
      non_fungible_id: e.non_fungible_id
    };
}
function Ji(e) {
  let t = !0;
  return t = t && "items" in e, t;
}
function T(e) {
  return ht(e);
}
function ht(e, t) {
  return e == null ? e : {
    total_count: i(e, "total_count") ? e.total_count : void 0,
    previous_cursor: i(e, "previous_cursor") ? e.previous_cursor : void 0,
    next_cursor: i(e, "next_cursor") ? e.next_cursor : void 0,
    items: e.items.map(Y)
  };
}
function w(e) {
  if (e !== void 0)
    return e === null ? null : {
      total_count: e.total_count,
      previous_cursor: e.previous_cursor,
      next_cursor: e.next_cursor,
      items: e.items.map(Z)
    };
}
function Ii(e) {
  let t = !0;
  return t = t && "ledger_state" in e, t = t && "address" in e, t = t && "resource_address" in e, t = t && "non_fungible_ids" in e, t;
}
function Ft(e) {
  return bt(e);
}
function bt(e, t) {
  return e == null ? e : {
    ledger_state: d(e.ledger_state),
    address: e.address,
    resource_address: e.resource_address,
    non_fungible_ids: T(e.non_fungible_ids)
  };
}
function Ei(e) {
  if (e !== void 0)
    return e === null ? null : {
      ledger_state: c(e.ledger_state),
      address: e.address,
      resource_address: e.resource_address,
      non_fungible_ids: w(e.non_fungible_ids)
    };
}
function qi(e) {
  let t = !0;
  return t = t && "address" in e, t = t && "resource_address" in e, t = t && "non_fungible_ids" in e, t;
}
function Di(e) {
  return Tt(e);
}
function Tt(e, t) {
  return e == null ? e : {
    address: e.address,
    resource_address: e.resource_address,
    non_fungible_ids: T(e.non_fungible_ids)
  };
}
function Ai(e) {
  if (e !== void 0)
    return e === null ? null : {
      address: e.address,
      resource_address: e.resource_address,
      non_fungible_ids: w(e.non_fungible_ids)
    };
}
function vi(e) {
  let t = !0;
  return t = t && "address" in e, t;
}
function xi(e) {
  return wt(e);
}
function wt(e, t) {
  return e == null ? e : {
    at_ledger_state: i(e, "at_ledger_state") ? f(e.at_ledger_state) : void 0,
    address: e.address,
    cursor: i(e, "cursor") ? e.cursor : void 0,
    limit: i(e, "limit") ? e.limit : void 0
  };
}
function Jt(e) {
  if (e !== void 0)
    return e === null ? null : {
      at_ledger_state: _(e.at_ledger_state),
      address: e.address,
      cursor: e.cursor,
      limit: e.limit
    };
}
function Ci(e) {
  let t = !0;
  return t = t && "address" in e, t = t && "amount" in e, t;
}
function j(e) {
  return It(e);
}
function It(e, t) {
  return e == null ? e : {
    address: e.address,
    amount: e.amount
  };
}
function ee(e) {
  if (e !== void 0)
    return e === null ? null : {
      address: e.address,
      amount: e.amount
    };
}
function ki(e) {
  let t = !0;
  return t = t && "items" in e, t;
}
function J(e) {
  return Et(e);
}
function Et(e, t) {
  return e == null ? e : {
    total_count: i(e, "total_count") ? e.total_count : void 0,
    previous_cursor: i(e, "previous_cursor") ? e.previous_cursor : void 0,
    next_cursor: i(e, "next_cursor") ? e.next_cursor : void 0,
    items: e.items.map(j)
  };
}
function I(e) {
  if (e !== void 0)
    return e === null ? null : {
      total_count: e.total_count,
      previous_cursor: e.previous_cursor,
      next_cursor: e.next_cursor,
      items: e.items.map(ee)
    };
}
function Pi(e) {
  let t = !0;
  return t = t && "ledger_state" in e, t = t && "address" in e, t = t && "non_fungibles" in e, t;
}
function qt(e) {
  return Dt(e);
}
function Dt(e, t) {
  return e == null ? e : {
    ledger_state: d(e.ledger_state),
    address: e.address,
    non_fungibles: J(e.non_fungibles)
  };
}
function Mi(e) {
  if (e !== void 0)
    return e === null ? null : {
      ledger_state: c(e.ledger_state),
      address: e.address,
      non_fungibles: I(e.non_fungibles)
    };
}
function Vi(e) {
  let t = !0;
  return t = t && "address" in e, t = t && "non_fungibles" in e, t;
}
function Li(e) {
  return At(e);
}
function At(e, t) {
  return e == null ? e : {
    address: e.address,
    non_fungibles: J(e.non_fungibles)
  };
}
function Ki(e) {
  if (e !== void 0)
    return e === null ? null : {
      address: e.address,
      non_fungibles: I(e.non_fungibles)
    };
}
function Gi(e) {
  let t = !0;
  return t = t && "type" in e, t;
}
function vt(e) {
  return y(e, !1);
}
function y(e, t) {
  if (e == null)
    return e;
  if (!t) {
    if (e.type === "EntityNotFoundError")
      return te(e, !0);
    if (e.type === "InternalServerError")
      return ce(e, !0);
    if (e.type === "InvalidEntityError")
      return le(e, !0);
    if (e.type === "InvalidRequestError")
      return me(e, !0);
    if (e.type === "InvalidTransactionError")
      return pe(e);
    if (e.type === "NotSyncedUpError")
      return ye(e, !0);
    if (e.type === "TransactionNotFoundError")
      return Te(e, !0);
  }
  return {
    type: e.type
  };
}
function g(e) {
  if (e !== void 0)
    return e === null ? null : {
      type: e.type
    };
}
function Ui(e) {
  let t = !0;
  return t = t && "address" in e, t;
}
function $i(e) {
  return te(e, !1);
}
function te(e, t) {
  return e == null ? e : {
    ...y(e, t),
    address: e.address
  };
}
function Bi(e) {
  if (e !== void 0)
    return e === null ? null : {
      ...g(e),
      address: e.address
    };
}
function zi(e) {
  let t = !0;
  return t = t && "addresses" in e, t;
}
function Hi(e) {
  return xt(e);
}
function xt(e, t) {
  return e == null ? e : {
    at_ledger_state: i(e, "at_ledger_state") ? f(e.at_ledger_state) : void 0,
    addresses: e.addresses
  };
}
function Ct(e) {
  if (e !== void 0)
    return e === null ? null : {
      at_ledger_state: _(e.at_ledger_state),
      addresses: e.addresses
    };
}
function Wi(e) {
  let t = !0;
  return t = t && "addresses" in e, t;
}
function Qi(e) {
  return kt(e);
}
function kt(e, t) {
  return e == null ? e : {
    addresses: e.addresses
  };
}
function Xi(e) {
  if (e !== void 0)
    return e === null ? null : {
      addresses: e.addresses
    };
}
function Yi(e) {
  let t = !0;
  return t = t && "address" in e, t = t && "metadata" in e, t;
}
function ne(e) {
  return Pt(e);
}
function Pt(e, t) {
  return e == null ? e : {
    address: e.address,
    metadata: m(e.metadata)
  };
}
function re(e) {
  if (e !== void 0)
    return e === null ? null : {
      address: e.address,
      metadata: p(e.metadata)
    };
}
function Zi(e) {
  let t = !0;
  return t = t && "ledger_state" in e, t = t && "entities" in e, t;
}
function Mt(e) {
  return Vt(e);
}
function Vt(e, t) {
  return e == null ? e : {
    ledger_state: d(e.ledger_state),
    entities: e.entities.map(ne)
  };
}
function ji(e) {
  if (e !== void 0)
    return e === null ? null : {
      ledger_state: c(e.ledger_state),
      entities: e.entities.map(re)
    };
}
function es(e) {
  let t = !0;
  return t = t && "entities" in e, t;
}
function ts(e) {
  return Lt(e);
}
function Lt(e, t) {
  return e == null ? e : {
    entities: e.entities.map(ne)
  };
}
function ns(e) {
  if (e !== void 0)
    return e === null ? null : {
      entities: e.entities.map(re)
    };
}
function rs(e) {
  let t = !0;
  return t = t && "address" in e, t;
}
function is(e) {
  return Kt(e);
}
function Kt(e, t) {
  return e == null ? e : {
    at_ledger_state: i(e, "at_ledger_state") ? f(e.at_ledger_state) : void 0,
    address: e.address
  };
}
function Gt(e) {
  if (e !== void 0)
    return e === null ? null : {
      at_ledger_state: _(e.at_ledger_state),
      address: e.address
    };
}
function ss(e) {
  let t = !0;
  return t = t && "address" in e, t;
}
function os(e) {
  return Ut(e);
}
function Ut(e, t) {
  return e == null ? e : {
    address: e.address
  };
}
function as(e) {
  if (e !== void 0)
    return e === null ? null : {
      address: e.address
    };
}
function us(e) {
  let t = !0;
  return t = t && "ledger_state" in e, t = t && "address" in e, t = t && "fungible_resources" in e, t = t && "non_fungible_resources" in e, t;
}
function $t(e) {
  return Bt(e);
}
function Bt(e, t) {
  return e == null ? e : {
    ledger_state: d(e.ledger_state),
    address: e.address,
    fungible_resources: F(e.fungible_resources),
    non_fungible_resources: J(e.non_fungible_resources)
  };
}
function ds(e) {
  if (e !== void 0)
    return e === null ? null : {
      ledger_state: c(e.ledger_state),
      address: e.address,
      fungible_resources: b(e.fungible_resources),
      non_fungible_resources: I(e.non_fungible_resources)
    };
}
function cs(e) {
  let t = !0;
  return t = t && "address" in e, t = t && "fungible_resources" in e, t = t && "non_fungible_resources" in e, t;
}
function ls(e) {
  return zt(e);
}
function zt(e, t) {
  return e == null ? e : {
    address: e.address,
    fungible_resources: F(e.fungible_resources),
    non_fungible_resources: J(e.non_fungible_resources)
  };
}
function fs(e) {
  if (e !== void 0)
    return e === null ? null : {
      address: e.address,
      fungible_resources: b(e.fungible_resources),
      non_fungible_resources: I(e.non_fungible_resources)
    };
}
function _s(e) {
  let t = !0;
  return t = t && "message" in e, t;
}
function ms(e) {
  return Ht(e);
}
function Ht(e, t) {
  return e == null ? e : {
    message: e.message,
    code: i(e, "code") ? e.code : void 0,
    details: i(e, "details") ? vt(e.details) : void 0,
    trace_id: i(e, "trace_id") ? e.trace_id : void 0
  };
}
function ps(e) {
  if (e !== void 0)
    return e === null ? null : {
      message: e.message,
      code: e.code,
      details: g(e.details),
      trace_id: e.trace_id
    };
}
function ys(e) {
  let t = !0;
  return t = t && "items" in e, t;
}
function gs(e) {
  return Wt(e);
}
function Wt(e, t) {
  return e == null ? e : {
    items: e.items.map(Q)
  };
}
function Os(e) {
  if (e !== void 0)
    return e === null ? null : {
      items: e.items.map(X)
    };
}
function Rs(e) {
  let t = !0;
  return t = t && "state_version" in e, t;
}
function ie(e) {
  return Qt(e);
}
function Qt(e, t) {
  return e == null ? e : {
    state_version: e.state_version
  };
}
function se(e) {
  if (e !== void 0)
    return e === null ? null : {
      state_version: e.state_version
    };
}
function Ss(e) {
  let t = !0;
  return t = t && "release_version" in e, t = t && "open_api_schema_version" in e, t;
}
function oe(e) {
  return Xt(e);
}
function Xt(e, t) {
  return e == null ? e : {
    release_version: e.release_version,
    open_api_schema_version: e.open_api_schema_version
  };
}
function ae(e) {
  if (e !== void 0)
    return e === null ? null : {
      release_version: e.release_version,
      open_api_schema_version: e.open_api_schema_version
    };
}
function Ns(e) {
  let t = !0;
  return t = t && "account_package" in e, t = t && "faucet" in e, t = t && "epoch_manager" in e, t = t && "clock" in e, t = t && "ecdsa_secp256k1" in e, t = t && "eddsa_ed25519" in e, t = t && "xrd" in e, t;
}
function ue(e) {
  return Yt(e);
}
function Yt(e, t) {
  return e == null ? e : {
    account_package: e.account_package,
    faucet: e.faucet,
    epoch_manager: e.epoch_manager,
    clock: e.clock,
    ecdsa_secp256k1: e.ecdsa_secp256k1,
    eddsa_ed25519: e.eddsa_ed25519,
    xrd: e.xrd
  };
}
function de(e) {
  if (e !== void 0)
    return e === null ? null : {
      account_package: e.account_package,
      faucet: e.faucet,
      epoch_manager: e.epoch_manager,
      clock: e.clock,
      ecdsa_secp256k1: e.ecdsa_secp256k1,
      eddsa_ed25519: e.eddsa_ed25519,
      xrd: e.xrd
    };
}
function hs(e) {
  let t = !0;
  return t = t && "ledger_state" in e, t = t && "known_target" in e, t = t && "release_info" in e, t = t && "well_known_addresses" in e, t;
}
function Zt(e) {
  return jt(e);
}
function jt(e, t) {
  return e == null ? e : {
    ledger_state: d(e.ledger_state),
    known_target: ie(e.known_target),
    release_info: oe(e.release_info),
    well_known_addresses: ue(e.well_known_addresses)
  };
}
function Fs(e) {
  if (e !== void 0)
    return e === null ? null : {
      ledger_state: c(e.ledger_state),
      known_target: se(e.known_target),
      release_info: ae(e.release_info),
      well_known_addresses: de(e.well_known_addresses)
    };
}
function bs(e) {
  let t = !0;
  return t = t && "known_target" in e, t = t && "release_info" in e, t = t && "well_known_addresses" in e, t;
}
function Ts(e) {
  return en(e);
}
function en(e, t) {
  return e == null ? e : {
    known_target: ie(e.known_target),
    release_info: oe(e.release_info),
    well_known_addresses: ue(e.well_known_addresses)
  };
}
function ws(e) {
  if (e !== void 0)
    return e === null ? null : {
      known_target: se(e.known_target),
      release_info: ae(e.release_info),
      well_known_addresses: de(e.well_known_addresses)
    };
}
function Js(e) {
  let t = !0;
  return t = t && "exception" in e, t = t && "cause" in e, t;
}
function Is(e) {
  return ce(e, !1);
}
function ce(e, t) {
  return e == null ? e : {
    ...y(e, t),
    exception: e.exception,
    cause: e.cause
  };
}
function Es(e) {
  if (e !== void 0)
    return e === null ? null : {
      ...g(e),
      exception: e.exception,
      cause: e.cause
    };
}
function qs(e) {
  let t = !0;
  return t = t && "exception" in e, t = t && "cause" in e, t;
}
function Ds(e) {
  return tn(e);
}
function tn(e, t) {
  return e == null ? e : {
    exception: e.exception,
    cause: e.cause
  };
}
function As(e) {
  if (e !== void 0)
    return e === null ? null : {
      exception: e.exception,
      cause: e.cause
    };
}
function vs(e) {
  let t = !0;
  return t = t && "address" in e, t;
}
function xs(e) {
  return le(e, !1);
}
function le(e, t) {
  return e == null ? e : {
    ...y(e, t),
    address: e.address
  };
}
function Cs(e) {
  if (e !== void 0)
    return e === null ? null : {
      ...g(e),
      address: e.address
    };
}
function ks(e) {
  let t = !0;
  return t = t && "path" in e, t = t && "errors" in e, t;
}
function fe(e) {
  return nn(e);
}
function nn(e, t) {
  return e == null ? e : {
    path: e.path,
    errors: e.errors
  };
}
function _e(e) {
  if (e !== void 0)
    return e === null ? null : {
      path: e.path,
      errors: e.errors
    };
}
function Ps(e) {
  let t = !0;
  return t = t && "validation_errors" in e, t;
}
function Ms(e) {
  return me(e, !1);
}
function me(e, t) {
  return e == null ? e : {
    ...y(e, t),
    validation_errors: e.validation_errors.map(fe)
  };
}
function Vs(e) {
  if (e !== void 0)
    return e === null ? null : {
      ...g(e),
      validation_errors: e.validation_errors.map(_e)
    };
}
function Ls(e) {
  let t = !0;
  return t = t && "validation_errors" in e, t;
}
function Ks(e) {
  return rn(e);
}
function rn(e, t) {
  return e == null ? e : {
    validation_errors: e.validation_errors.map(fe)
  };
}
function Gs(e) {
  if (e !== void 0)
    return e === null ? null : {
      validation_errors: e.validation_errors.map(_e)
    };
}
function Us(e) {
  return !0;
}
function $s(e) {
  return pe(e);
}
function pe(e, t) {
  return e;
}
function Bs(e) {
  return e;
}
function zs(e) {
  let t = !0;
  return t = t && "ledger_state" in e, t;
}
function Hs(e) {
  return sn(e);
}
function sn(e, t) {
  return e == null ? e : {
    ledger_state: d(e.ledger_state)
  };
}
function Ws(e) {
  if (e !== void 0)
    return e === null ? null : {
      ledger_state: c(e.ledger_state)
    };
}
function Qs(e) {
  let t = !0;
  return t = t && "address" in e, t = t && "non_fungible_id" in e, t;
}
function Xs(e) {
  return on(e);
}
function on(e, t) {
  return e == null ? e : {
    at_ledger_state: i(e, "at_ledger_state") ? f(e.at_ledger_state) : void 0,
    address: e.address,
    non_fungible_id: e.non_fungible_id,
    cursor: i(e, "cursor") ? e.cursor : void 0,
    limit: i(e, "limit") ? e.limit : void 0
  };
}
function an(e) {
  if (e !== void 0)
    return e === null ? null : {
      at_ledger_state: _(e.at_ledger_state),
      address: e.address,
      non_fungible_id: e.non_fungible_id,
      cursor: e.cursor,
      limit: e.limit
    };
}
function Ys(e) {
  let t = !0;
  return t = t && "address" in e, t = t && "non_fungible_id" in e, t;
}
function Zs(e) {
  return un(e);
}
function un(e, t) {
  return e == null ? e : {
    address: e.address,
    non_fungible_id: e.non_fungible_id,
    cursor: i(e, "cursor") ? e.cursor : void 0,
    limit: i(e, "limit") ? e.limit : void 0
  };
}
function js(e) {
  if (e !== void 0)
    return e === null ? null : {
      address: e.address,
      non_fungible_id: e.non_fungible_id,
      cursor: e.cursor,
      limit: e.limit
    };
}
function eo(e) {
  let t = !0;
  return t = t && "ledger_state" in e, t = t && "address" in e, t = t && "non_fungible_id_type" in e, t = t && "non_fungible_id" in e, t = t && "mutable_data_hex" in e, t = t && "immutable_data_hex" in e, t;
}
function dn(e) {
  return cn(e);
}
function cn(e, t) {
  return e == null ? e : {
    ledger_state: d(e.ledger_state),
    address: e.address,
    non_fungible_id_type: C(e.non_fungible_id_type),
    non_fungible_id: e.non_fungible_id,
    mutable_data_hex: e.mutable_data_hex,
    immutable_data_hex: e.immutable_data_hex
  };
}
function to(e) {
  if (e !== void 0)
    return e === null ? null : {
      ledger_state: c(e.ledger_state),
      address: e.address,
      non_fungible_id_type: e.non_fungible_id_type,
      non_fungible_id: e.non_fungible_id,
      mutable_data_hex: e.mutable_data_hex,
      immutable_data_hex: e.immutable_data_hex
    };
}
function no(e) {
  let t = !0;
  return t = t && "address" in e, t = t && "non_fungible_id_type" in e, t = t && "non_fungible_id" in e, t = t && "mutable_data_hex" in e, t = t && "immutable_data_hex" in e, t;
}
function ro(e) {
  return ln(e);
}
function ln(e, t) {
  return e == null ? e : {
    address: e.address,
    non_fungible_id_type: C(e.non_fungible_id_type),
    non_fungible_id: e.non_fungible_id,
    mutable_data_hex: e.mutable_data_hex,
    immutable_data_hex: e.immutable_data_hex
  };
}
function io(e) {
  if (e !== void 0)
    return e === null ? null : {
      address: e.address,
      non_fungible_id_type: e.non_fungible_id_type,
      non_fungible_id: e.non_fungible_id,
      mutable_data_hex: e.mutable_data_hex,
      immutable_data_hex: e.immutable_data_hex
    };
}
function so(e) {
  let t = !0;
  return t = t && "items" in e, t;
}
function oo(e) {
  return fn(e);
}
function fn(e, t) {
  return e == null ? e : {
    items: e.items.map(Y)
  };
}
function ao(e) {
  if (e !== void 0)
    return e === null ? null : {
      items: e.items.map(Z)
    };
}
function uo(e) {
  let t = !0;
  return t = t && "address" in e, t;
}
function co(e) {
  return _n(e);
}
function _n(e, t) {
  return e == null ? e : {
    at_ledger_state: i(e, "at_ledger_state") ? f(e.at_ledger_state) : void 0,
    address: e.address,
    cursor: i(e, "cursor") ? e.cursor : void 0,
    limit: i(e, "limit") ? e.limit : void 0
  };
}
function mn(e) {
  if (e !== void 0)
    return e === null ? null : {
      at_ledger_state: _(e.at_ledger_state),
      address: e.address,
      cursor: e.cursor,
      limit: e.limit
    };
}
function lo(e) {
  let t = !0;
  return t = t && "ledger_state" in e, t = t && "address" in e, t = t && "non_fungible_ids" in e, t;
}
function pn(e) {
  return yn(e);
}
function yn(e, t) {
  return e == null ? e : {
    ledger_state: d(e.ledger_state),
    address: e.address,
    non_fungible_ids: T(e.non_fungible_ids)
  };
}
function fo(e) {
  if (e !== void 0)
    return e === null ? null : {
      ledger_state: c(e.ledger_state),
      address: e.address,
      non_fungible_ids: w(e.non_fungible_ids)
    };
}
function _o(e) {
  let t = !0;
  return t = t && "address" in e, t = t && "non_fungible_ids" in e, t;
}
function mo(e) {
  return gn(e);
}
function gn(e, t) {
  return e == null ? e : {
    address: e.address,
    non_fungible_ids: T(e.non_fungible_ids)
  };
}
function po(e) {
  if (e !== void 0)
    return e === null ? null : {
      address: e.address,
      non_fungible_ids: w(e.non_fungible_ids)
    };
}
function yo(e) {
  let t = !0;
  return t = t && "items" in e, t;
}
function go(e) {
  return On(e);
}
function On(e, t) {
  return e == null ? e : {
    items: e.items.map(j)
  };
}
function Oo(e) {
  if (e !== void 0)
    return e === null ? null : {
      items: e.items.map(ee)
    };
}
function Ro(e) {
  let t = !0;
  return t = t && "request_type" in e, t = t && "current_sync_delay_seconds" in e, t = t && "max_allowed_sync_delay_seconds" in e, t;
}
function So(e) {
  return ye(e, !1);
}
function ye(e, t) {
  return e == null ? e : {
    ...y(e, t),
    request_type: e.request_type,
    current_sync_delay_seconds: e.current_sync_delay_seconds,
    max_allowed_sync_delay_seconds: e.max_allowed_sync_delay_seconds
  };
}
function No(e) {
  if (e !== void 0)
    return e === null ? null : {
      ...g(e),
      request_type: e.request_type,
      current_sync_delay_seconds: e.current_sync_delay_seconds,
      max_allowed_sync_delay_seconds: e.max_allowed_sync_delay_seconds
    };
}
function ho(e) {
  let t = !0;
  return t = t && "request_type" in e, t = t && "current_sync_delay_seconds" in e, t = t && "max_allowed_sync_delay_seconds" in e, t;
}
function Fo(e) {
  return Rn(e);
}
function Rn(e, t) {
  return e == null ? e : {
    request_type: e.request_type,
    current_sync_delay_seconds: e.current_sync_delay_seconds,
    max_allowed_sync_delay_seconds: e.max_allowed_sync_delay_seconds
  };
}
function bo(e) {
  if (e !== void 0)
    return e === null ? null : {
      request_type: e.request_type,
      current_sync_delay_seconds: e.current_sync_delay_seconds,
      max_allowed_sync_delay_seconds: e.max_allowed_sync_delay_seconds
    };
}
const To = {
  EcdsaSecp256k1: "EcdsaSecp256k1",
  EddsaEd25519: "EddsaEd25519"
};
function Sn(e) {
  return Nn(e);
}
function Nn(e, t) {
  return e;
}
function wo(e) {
  return e;
}
function Jo(e) {
  let t = !0;
  return t = t && "key_type" in e, t;
}
function hn(e) {
  return k(e, !1);
}
function k(e, t) {
  if (e == null)
    return e;
  if (!t) {
    if (e.key_type === "EcdsaSecp256k1")
      return ge(e, !0);
    if (e.key_type === "EddsaEd25519")
      return Oe(e, !0);
  }
  return {
    key_type: Sn(e.key_type)
  };
}
function P(e) {
  if (e !== void 0)
    return e === null ? null : {
      key_type: e.key_type
    };
}
function Io(e) {
  let t = !0;
  return t = t && "key_hex" in e, t;
}
function Eo(e) {
  return ge(e, !1);
}
function ge(e, t) {
  return e == null ? e : {
    ...k(e, t),
    key_hex: e.key_hex
  };
}
function qo(e) {
  if (e !== void 0)
    return e === null ? null : {
      ...P(e),
      key_hex: e.key_hex
    };
}
function Do(e) {
  let t = !0;
  return t = t && "key_hex" in e, t;
}
function Ao(e) {
  return Fn(e);
}
function Fn(e, t) {
  return e == null ? e : {
    key_hex: e.key_hex
  };
}
function vo(e) {
  if (e !== void 0)
    return e === null ? null : {
      key_hex: e.key_hex
    };
}
function xo(e) {
  let t = !0;
  return t = t && "key_hex" in e, t;
}
function Co(e) {
  return Oe(e, !1);
}
function Oe(e, t) {
  return e == null ? e : {
    ...k(e, t),
    key_hex: e.key_hex
  };
}
function ko(e) {
  if (e !== void 0)
    return e === null ? null : {
      ...P(e),
      key_hex: e.key_hex
    };
}
function Po(e) {
  let t = !0;
  return t = t && "key_hex" in e, t;
}
function Mo(e) {
  return bn(e);
}
function bn(e, t) {
  return e == null ? e : {
    key_hex: e.key_hex
  };
}
function Vo(e) {
  if (e !== void 0)
    return e === null ? null : {
      key_hex: e.key_hex
    };
}
function Lo(e) {
  return !0;
}
function Ko(e) {
  return Tn(e);
}
function Tn(e, t) {
  return e == null ? e : {
    total_count: i(e, "total_count") ? e.total_count : void 0,
    previous_cursor: i(e, "previous_cursor") ? e.previous_cursor : void 0,
    next_cursor: i(e, "next_cursor") ? e.next_cursor : void 0
  };
}
function Go(e) {
  if (e !== void 0)
    return e === null ? null : {
      total_count: e.total_count,
      previous_cursor: e.previous_cursor,
      next_cursor: e.next_cursor
    };
}
function Uo(e) {
  return !0;
}
function $o(e) {
  return wn(e);
}
function wn(e, t) {
  return e == null ? e : {
    at_ledger_state: i(e, "at_ledger_state") ? f(e.at_ledger_state) : void 0,
    cursor: i(e, "cursor") ? e.cursor : void 0
  };
}
function Jn(e) {
  if (e !== void 0)
    return e === null ? null : {
      at_ledger_state: _(e.at_ledger_state),
      cursor: e.cursor
    };
}
function Bo(e) {
  return !0;
}
function zo(e) {
  return In(e);
}
function In(e, t) {
  return e == null ? e : {
    cursor: i(e, "cursor") ? e.cursor : void 0
  };
}
function Ho(e) {
  if (e !== void 0)
    return e === null ? null : {
      cursor: e.cursor
    };
}
function Wo(e) {
  let t = !0;
  return t = t && "stake" in e, t = t && "stake_percentage" in e, t;
}
function En(e) {
  return qn(e);
}
function qn(e, t) {
  return e == null ? e : {
    stake: D(e.stake),
    stake_percentage: e.stake_percentage
  };
}
function Dn(e) {
  if (e !== void 0)
    return e === null ? null : {
      stake: A(e.stake),
      stake_percentage: e.stake_percentage
    };
}
function Qo(e) {
  let t = !0;
  return t = t && "stake" in e, t = t && "key" in e, t;
}
function An(e) {
  return vn(e);
}
function vn(e, t) {
  return e == null ? e : {
    stake: En(e.stake),
    key: hn(e.key),
    metadata: i(e, "metadata") ? m(e.metadata) : void 0
  };
}
function xn(e) {
  if (e !== void 0)
    return e === null ? null : {
      stake: Dn(e.stake),
      key: P(e.key),
      metadata: p(e.metadata)
    };
}
function Xo(e) {
  let t = !0;
  return t = t && "address" in e, t = t && "metadata" in e, t;
}
function Re(e) {
  return Cn(e);
}
function Cn(e, t) {
  return e == null ? e : {
    address: e.address,
    state: i(e, "state") ? e.state : void 0,
    active_in_epoch: i(e, "active_in_epoch") ? An(e.active_in_epoch) : void 0,
    metadata: m(e.metadata)
  };
}
function Se(e) {
  if (e !== void 0)
    return e === null ? null : {
      address: e.address,
      state: e.state,
      active_in_epoch: xn(e.active_in_epoch),
      metadata: p(e.metadata)
    };
}
function Yo(e) {
  let t = !0;
  return t = t && "items" in e, t;
}
function Ne(e) {
  return kn(e);
}
function kn(e, t) {
  return e == null ? e : {
    total_count: i(e, "total_count") ? e.total_count : void 0,
    previous_cursor: i(e, "previous_cursor") ? e.previous_cursor : void 0,
    next_cursor: i(e, "next_cursor") ? e.next_cursor : void 0,
    items: e.items.map(Re)
  };
}
function he(e) {
  if (e !== void 0)
    return e === null ? null : {
      total_count: e.total_count,
      previous_cursor: e.previous_cursor,
      next_cursor: e.next_cursor,
      items: e.items.map(Se)
    };
}
function Zo(e) {
  let t = !0;
  return t = t && "ledger_state" in e, t = t && "validators" in e, t;
}
function Pn(e) {
  return Mn(e);
}
function Mn(e, t) {
  return e == null ? e : {
    ledger_state: d(e.ledger_state),
    validators: Ne(e.validators)
  };
}
function jo(e) {
  if (e !== void 0)
    return e === null ? null : {
      ledger_state: c(e.ledger_state),
      validators: he(e.validators)
    };
}
function ea(e) {
  let t = !0;
  return t = t && "validators" in e, t;
}
function ta(e) {
  return Vn(e);
}
function Vn(e, t) {
  return e == null ? e : {
    validators: Ne(e.validators)
  };
}
function na(e) {
  if (e !== void 0)
    return e === null ? null : {
      validators: he(e.validators)
    };
}
const ra = {
  IntentHash: "intent_hash",
  SignedIntentHash: "signed_intent_hash",
  PayloadHash: "payload_hash"
};
function Ln(e) {
  return Kn(e);
}
function Kn(e, t) {
  return e;
}
function ia(e) {
  return e;
}
function sa(e) {
  let t = !0;
  return t = t && "type" in e, t = t && "value_hex" in e, t;
}
function E(e) {
  return Gn(e);
}
function Gn(e, t) {
  return e == null ? e : {
    type: Ln(e.type),
    value_hex: e.value_hex
  };
}
function q(e) {
  if (e !== void 0)
    return e === null ? null : {
      type: e.type,
      value_hex: e.value_hex
    };
}
function oa(e) {
  let t = !0;
  return t = t && "transaction_identifier" in e, t;
}
function aa(e) {
  return Un(e);
}
function Un(e, t) {
  return e == null ? e : {
    at_ledger_state: i(e, "at_ledger_state") ? f(e.at_ledger_state) : void 0,
    transaction_identifier: E(e.transaction_identifier)
  };
}
function $n(e) {
  if (e !== void 0)
    return e === null ? null : {
      at_ledger_state: _(e.at_ledger_state),
      transaction_identifier: q(e.transaction_identifier)
    };
}
function ua(e) {
  let t = !0;
  return t = t && "transaction_identifier" in e, t;
}
function da(e) {
  return Bn(e);
}
function Bn(e, t) {
  return e == null ? e : {
    transaction_identifier: E(e.transaction_identifier)
  };
}
function ca(e) {
  if (e !== void 0)
    return e === null ? null : {
      transaction_identifier: q(e.transaction_identifier)
    };
}
function la(e) {
  let t = !0;
  return t = t && "raw_hex" in e, t = t && "receipt" in e, t = t && "referenced_global_entities" in e, t;
}
function Fe(e) {
  return zn(e);
}
function zn(e, t) {
  return e == null ? e : {
    raw_hex: e.raw_hex,
    receipt: e.receipt,
    referenced_global_entities: e.referenced_global_entities,
    message_hex: i(e, "message_hex") ? e.message_hex : void 0
  };
}
function be(e) {
  if (e !== void 0)
    return e === null ? null : {
      raw_hex: e.raw_hex,
      receipt: e.receipt,
      referenced_global_entities: e.referenced_global_entities,
      message_hex: e.message_hex
    };
}
function fa(e) {
  let t = !0;
  return t = t && "ledger_state" in e, t = t && "transaction" in e, t = t && "details" in e, t;
}
function Hn(e) {
  return Wn(e);
}
function Wn(e, t) {
  return e == null ? e : {
    ledger_state: d(e.ledger_state),
    transaction: v(e.transaction),
    details: Fe(e.details)
  };
}
function _a(e) {
  if (e !== void 0)
    return e === null ? null : {
      ledger_state: c(e.ledger_state),
      transaction: x(e.transaction),
      details: be(e.details)
    };
}
function ma(e) {
  let t = !0;
  return t = t && "transaction" in e, t = t && "details" in e, t;
}
function pa(e) {
  return Qn(e);
}
function Qn(e, t) {
  return e == null ? e : {
    transaction: v(e.transaction),
    details: Fe(e.details)
  };
}
function ya(e) {
  if (e !== void 0)
    return e === null ? null : {
      transaction: x(e.transaction),
      details: be(e.details)
    };
}
function ga(e) {
  let t = !0;
  return t = t && "ledger_state" in e, t;
}
function Xn(e) {
  return Yn(e);
}
function Yn(e, t) {
  return e == null ? e : {
    ledger_state: d(e.ledger_state)
  };
}
function Oa(e) {
  if (e !== void 0)
    return e === null ? null : {
      ledger_state: c(e.ledger_state)
    };
}
function Ra(e) {
  let t = !0;
  return t = t && "transaction_not_found" in e, t;
}
function Sa(e) {
  return Te(e, !1);
}
function Te(e, t) {
  return e == null ? e : {
    ...y(e, t),
    transaction_not_found: E(e.transaction_not_found)
  };
}
function Na(e) {
  if (e !== void 0)
    return e === null ? null : {
      ...g(e),
      transaction_not_found: q(e.transaction_not_found)
    };
}
function ha(e) {
  let t = !0;
  return t = t && "transaction_not_found" in e, t;
}
function Fa(e) {
  return Zn(e);
}
function Zn(e, t) {
  return e == null ? e : {
    transaction_not_found: E(e.transaction_not_found)
  };
}
function ba(e) {
  if (e !== void 0)
    return e === null ? null : {
      transaction_not_found: q(e.transaction_not_found)
    };
}
function Ta(e) {
  return !0;
}
function wa(e) {
  return jn(e);
}
function jn(e, t) {
  return e == null ? e : {
    at_ledger_state: i(e, "at_ledger_state") ? f(e.at_ledger_state) : void 0,
    from_ledger_state: i(e, "from_ledger_state") ? f(e.from_ledger_state) : void 0,
    cursor: i(e, "cursor") ? e.cursor : void 0,
    limit: i(e, "limit") ? e.limit : void 0
  };
}
function er(e) {
  if (e !== void 0)
    return e === null ? null : {
      at_ledger_state: _(e.at_ledger_state),
      from_ledger_state: _(e.from_ledger_state),
      cursor: e.cursor,
      limit: e.limit
    };
}
function Ja(e) {
  return !0;
}
function Ia(e) {
  return tr(e);
}
function tr(e, t) {
  return e == null ? e : {
    from_ledger_state: i(e, "from_ledger_state") ? f(e.from_ledger_state) : void 0,
    cursor: i(e, "cursor") ? e.cursor : void 0,
    limit: i(e, "limit") ? e.limit : void 0
  };
}
function Ea(e) {
  if (e !== void 0)
    return e === null ? null : {
      from_ledger_state: _(e.from_ledger_state),
      cursor: e.cursor,
      limit: e.limit
    };
}
function qa(e) {
  let t = !0;
  return t = t && "ledger_state" in e, t = t && "items" in e, t;
}
function nr(e) {
  return rr(e);
}
function rr(e, t) {
  return e == null ? e : {
    ledger_state: d(e.ledger_state),
    total_count: i(e, "total_count") ? e.total_count : void 0,
    previous_cursor: i(e, "previous_cursor") ? e.previous_cursor : void 0,
    next_cursor: i(e, "next_cursor") ? e.next_cursor : void 0,
    items: e.items.map(v)
  };
}
function Da(e) {
  if (e !== void 0)
    return e === null ? null : {
      ledger_state: c(e.ledger_state),
      total_count: e.total_count,
      previous_cursor: e.previous_cursor,
      next_cursor: e.next_cursor,
      items: e.items.map(x)
    };
}
function Aa(e) {
  let t = !0;
  return t = t && "intent_hash_hex" in e, t;
}
function va(e) {
  return ir(e);
}
function ir(e, t) {
  return e == null ? e : {
    intent_hash_hex: e.intent_hash_hex
  };
}
function sr(e) {
  if (e !== void 0)
    return e === null ? null : {
      intent_hash_hex: e.intent_hash_hex
    };
}
function xa(e) {
  let t = !0;
  return t = t && "intent_hash_hex" in e, t;
}
function Ca(e) {
  return or(e);
}
function or(e, t) {
  return e == null ? e : {
    intent_hash_hex: e.intent_hash_hex
  };
}
function ka(e) {
  if (e !== void 0)
    return e === null ? null : {
      intent_hash_hex: e.intent_hash_hex
    };
}
function Pa(e) {
  let t = !0;
  return t = t && "payload_hash_hex" in e, t = t && "status" in e, t;
}
function we(e) {
  return ar(e);
}
function ar(e, t) {
  return e == null ? e : {
    payload_hash_hex: e.payload_hash_hex,
    status: N(e.status),
    error_message: i(e, "error_message") ? e.error_message : void 0
  };
}
function Je(e) {
  if (e !== void 0)
    return e === null ? null : {
      payload_hash_hex: e.payload_hash_hex,
      status: e.status,
      error_message: e.error_message
    };
}
function Ma(e) {
  let t = !0;
  return t = t && "ledger_state" in e, t = t && "status" in e, t = t && "known_payloads" in e, t;
}
function ur(e) {
  return dr(e);
}
function dr(e, t) {
  return e == null ? e : {
    ledger_state: d(e.ledger_state),
    status: N(e.status),
    known_payloads: e.known_payloads.map(we),
    error_message: i(e, "error_message") ? e.error_message : void 0
  };
}
function Va(e) {
  if (e !== void 0)
    return e === null ? null : {
      ledger_state: c(e.ledger_state),
      status: e.status,
      known_payloads: e.known_payloads.map(Je),
      error_message: e.error_message
    };
}
function La(e) {
  let t = !0;
  return t = t && "status" in e, t = t && "known_payloads" in e, t;
}
function Ka(e) {
  return cr(e);
}
function cr(e, t) {
  return e == null ? e : {
    status: N(e.status),
    known_payloads: e.known_payloads.map(we),
    error_message: i(e, "error_message") ? e.error_message : void 0
  };
}
function Ga(e) {
  if (e !== void 0)
    return e === null ? null : {
      status: e.status,
      known_payloads: e.known_payloads.map(Je),
      error_message: e.error_message
    };
}
function Ua(e) {
  let t = !0;
  return t = t && "notarized_transaction_hex" in e, t;
}
function $a(e) {
  return lr(e);
}
function lr(e, t) {
  return e == null ? e : {
    notarized_transaction_hex: e.notarized_transaction_hex
  };
}
function fr(e) {
  if (e !== void 0)
    return e === null ? null : {
      notarized_transaction_hex: e.notarized_transaction_hex
    };
}
function Ba(e) {
  let t = !0;
  return t = t && "duplicate" in e, t;
}
function _r(e) {
  return mr(e);
}
function mr(e, t) {
  return e == null ? e : {
    duplicate: e.duplicate
  };
}
function za(e) {
  if (e !== void 0)
    return e === null ? null : {
      duplicate: e.duplicate
    };
}
function Ha(e) {
  let t = !0;
  return t = t && "items" in e, t;
}
function Wa(e) {
  return pr(e);
}
function pr(e, t) {
  return e == null ? e : {
    items: e.items.map(Re)
  };
}
function Qa(e) {
  if (e !== void 0)
    return e === null ? null : {
      items: e.items.map(Se)
    };
}
class Xa extends S {
  async entityDetailsRaw(t, n) {
    if (t.entityDetailsRequest === null || t.entityDetailsRequest === void 0)
      throw new l("entityDetailsRequest", "Required parameter requestParameters.entityDetailsRequest was null or undefined when calling entityDetails.");
    const r = {}, s = {};
    s["Content-Type"] = "application/json";
    const o = await this.request({
      path: "/entity/details",
      method: "POST",
      headers: s,
      query: r,
      body: $e(t.entityDetailsRequest)
    }, n);
    return new u(o, (a) => tt(a));
  }
  async entityDetails(t, n) {
    return await (await this.entityDetailsRaw(t, n)).value();
  }
  async entityFungiblesRaw(t, n) {
    if (t.entityFungiblesRequest === null || t.entityFungiblesRequest === void 0)
      throw new l("entityFungiblesRequest", "Required parameter requestParameters.entityFungiblesRequest was null or undefined when calling entityFungibles.");
    const r = {}, s = {};
    s["Content-Type"] = "application/json";
    const o = await this.request({
      path: "/entity/fungibles",
      method: "POST",
      headers: s,
      query: r,
      body: st(t.entityFungiblesRequest)
    }, n);
    return new u(o, (a) => ut(a));
  }
  async entityFungibles(t, n) {
    return await (await this.entityFungiblesRaw(t, n)).value();
  }
  async entityMetadataRaw(t, n) {
    if (t.entityMetadataRequest === null || t.entityMetadataRequest === void 0)
      throw new l("entityMetadataRequest", "Required parameter requestParameters.entityMetadataRequest was null or undefined when calling entityMetadata.");
    const r = {}, s = {};
    s["Content-Type"] = "application/json";
    const o = await this.request({
      path: "/entity/metadata",
      method: "POST",
      headers: s,
      query: r,
      body: _t(t.entityMetadataRequest)
    }, n);
    return new u(o, (a) => pt(a));
  }
  async entityMetadata(t, n) {
    return await (await this.entityMetadataRaw(t, n)).value();
  }
  async entityNonFungibleIdsRaw(t, n) {
    if (t.entityNonFungibleIdsRequest === null || t.entityNonFungibleIdsRequest === void 0)
      throw new l("entityNonFungibleIdsRequest", "Required parameter requestParameters.entityNonFungibleIdsRequest was null or undefined when calling entityNonFungibleIds.");
    const r = {}, s = {};
    s["Content-Type"] = "application/json";
    const o = await this.request({
      path: "/entity/non-fungible/ids",
      method: "POST",
      headers: s,
      query: r,
      body: Rt(t.entityNonFungibleIdsRequest)
    }, n);
    return new u(o, (a) => Ft(a));
  }
  async entityNonFungibleIds(t, n) {
    return await (await this.entityNonFungibleIdsRaw(t, n)).value();
  }
  async entityNonFungiblesRaw(t, n) {
    if (t.entityNonFungiblesRequest === null || t.entityNonFungiblesRequest === void 0)
      throw new l("entityNonFungiblesRequest", "Required parameter requestParameters.entityNonFungiblesRequest was null or undefined when calling entityNonFungibles.");
    const r = {}, s = {};
    s["Content-Type"] = "application/json";
    const o = await this.request({
      path: "/entity/non-fungibles",
      method: "POST",
      headers: s,
      query: r,
      body: Jt(t.entityNonFungiblesRequest)
    }, n);
    return new u(o, (a) => qt(a));
  }
  async entityNonFungibles(t, n) {
    return await (await this.entityNonFungiblesRaw(t, n)).value();
  }
  async entityOverviewRaw(t, n) {
    if (t.entityOverviewRequest === null || t.entityOverviewRequest === void 0)
      throw new l("entityOverviewRequest", "Required parameter requestParameters.entityOverviewRequest was null or undefined when calling entityOverview.");
    const r = {}, s = {};
    s["Content-Type"] = "application/json";
    const o = await this.request({
      path: "/entity/overview",
      method: "POST",
      headers: s,
      query: r,
      body: Ct(t.entityOverviewRequest)
    }, n);
    return new u(o, (a) => Mt(a));
  }
  async entityOverview(t, n) {
    return await (await this.entityOverviewRaw(t, n)).value();
  }
  async entityResourcesRaw(t, n) {
    if (t.entityResourcesRequest === null || t.entityResourcesRequest === void 0)
      throw new l("entityResourcesRequest", "Required parameter requestParameters.entityResourcesRequest was null or undefined when calling entityResources.");
    const r = {}, s = {};
    s["Content-Type"] = "application/json";
    const o = await this.request({
      path: "/entity/resources",
      method: "POST",
      headers: s,
      query: r,
      body: Gt(t.entityResourcesRequest)
    }, n);
    return new u(o, (a) => $t(a));
  }
  async entityResources(t, n) {
    return await (await this.entityResourcesRaw(t, n)).value();
  }
  async nonFungibleIdDataRaw(t, n) {
    if (t.nonFungibleDataRequest === null || t.nonFungibleDataRequest === void 0)
      throw new l("nonFungibleDataRequest", "Required parameter requestParameters.nonFungibleDataRequest was null or undefined when calling nonFungibleIdData.");
    const r = {}, s = {};
    s["Content-Type"] = "application/json";
    const o = await this.request({
      path: "/non-fungible/data",
      method: "POST",
      headers: s,
      query: r,
      body: an(t.nonFungibleDataRequest)
    }, n);
    return new u(o, (a) => dn(a));
  }
  async nonFungibleIdData(t, n) {
    return await (await this.nonFungibleIdDataRaw(t, n)).value();
  }
  async nonFungibleIdsRaw(t, n) {
    if (t.nonFungibleIdsRequest === null || t.nonFungibleIdsRequest === void 0)
      throw new l("nonFungibleIdsRequest", "Required parameter requestParameters.nonFungibleIdsRequest was null or undefined when calling nonFungibleIds.");
    const r = {}, s = {};
    s["Content-Type"] = "application/json";
    const o = await this.request({
      path: "/non-fungible/ids",
      method: "POST",
      headers: s,
      query: r,
      body: mn(t.nonFungibleIdsRequest)
    }, n);
    return new u(o, (a) => pn(a));
  }
  async nonFungibleIds(t, n) {
    return await (await this.nonFungibleIdsRaw(t, n)).value();
  }
  async stateValidatorsListRaw(t, n) {
    if (t.stateValidatorsListRequest === null || t.stateValidatorsListRequest === void 0)
      throw new l("stateValidatorsListRequest", "Required parameter requestParameters.stateValidatorsListRequest was null or undefined when calling stateValidatorsList.");
    const r = {}, s = {};
    s["Content-Type"] = "application/json";
    const o = await this.request({
      path: "/state/validators/list",
      method: "POST",
      headers: s,
      query: r,
      body: Jn(t.stateValidatorsListRequest)
    }, n);
    return new u(o, (a) => Pn(a));
  }
  async stateValidatorsList(t, n) {
    return await (await this.stateValidatorsListRaw(t, n)).value();
  }
}
class Ya extends S {
  async gatewayInformationRaw(t) {
    const n = {}, r = {}, s = await this.request({
      path: "/gateway/information",
      method: "POST",
      headers: r,
      query: n
    }, t);
    return new u(s, (o) => Zt(o));
  }
  async gatewayInformation(t) {
    return await (await this.gatewayInformationRaw(t)).value();
  }
}
class Za extends S {
  async transactionRecentRaw(t, n) {
    if (t.transactionRecentRequest === null || t.transactionRecentRequest === void 0)
      throw new l("transactionRecentRequest", "Required parameter requestParameters.transactionRecentRequest was null or undefined when calling transactionRecent.");
    const r = {}, s = {};
    s["Content-Type"] = "application/json";
    const o = await this.request({
      path: "/transaction/recent",
      method: "POST",
      headers: s,
      query: r,
      body: er(t.transactionRecentRequest)
    }, n);
    return new u(o, (a) => nr(a));
  }
  async transactionRecent(t, n) {
    return await (await this.transactionRecentRaw(t, n)).value();
  }
}
class ja extends S {
  async transactionCommittedDetailsRaw(t, n) {
    if (t.transactionCommittedDetailsRequest === null || t.transactionCommittedDetailsRequest === void 0)
      throw new l("transactionCommittedDetailsRequest", "Required parameter requestParameters.transactionCommittedDetailsRequest was null or undefined when calling transactionCommittedDetails.");
    const r = {}, s = {};
    s["Content-Type"] = "application/json";
    const o = await this.request({
      path: "/transaction/committed-details",
      method: "POST",
      headers: s,
      query: r,
      body: $n(t.transactionCommittedDetailsRequest)
    }, n);
    return new u(o, (a) => Hn(a));
  }
  async transactionCommittedDetails(t, n) {
    return await (await this.transactionCommittedDetailsRaw(t, n)).value();
  }
  async transactionConstructionRaw(t) {
    const n = {}, r = {}, s = await this.request({
      path: "/transaction/construction",
      method: "POST",
      headers: r,
      query: n
    }, t);
    return new u(s, (o) => Xn(o));
  }
  async transactionConstruction(t) {
    return await (await this.transactionConstructionRaw(t)).value();
  }
  async transactionPreviewRaw(t, n) {
    if (t.body === null || t.body === void 0)
      throw new l("body", "Required parameter requestParameters.body was null or undefined when calling transactionPreview.");
    const r = {}, s = {};
    s["Content-Type"] = "application/json";
    const o = await this.request({
      path: "/transaction/preview",
      method: "POST",
      headers: s,
      query: r,
      body: t.body
    }, n);
    return new u(o);
  }
  async transactionPreview(t, n) {
    return await (await this.transactionPreviewRaw(t, n)).value();
  }
  async transactionStatusRaw(t, n) {
    if (t.transactionStatusRequest === null || t.transactionStatusRequest === void 0)
      throw new l("transactionStatusRequest", "Required parameter requestParameters.transactionStatusRequest was null or undefined when calling transactionStatus.");
    const r = {}, s = {};
    s["Content-Type"] = "application/json";
    const o = await this.request({
      path: "/transaction/status",
      method: "POST",
      headers: s,
      query: r,
      body: sr(t.transactionStatusRequest)
    }, n);
    return new u(o, (a) => ur(a));
  }
  async transactionStatus(t, n) {
    return await (await this.transactionStatusRaw(t, n)).value();
  }
  async transactionSubmitRaw(t, n) {
    if (t.transactionSubmitRequest === null || t.transactionSubmitRequest === void 0)
      throw new l("transactionSubmitRequest", "Required parameter requestParameters.transactionSubmitRequest was null or undefined when calling transactionSubmit.");
    const r = {}, s = {};
    s["Content-Type"] = "application/json";
    const o = await this.request({
      path: "/transaction/submit",
      method: "POST",
      headers: s,
      query: r,
      body: fr(t.transactionSubmitRequest)
    }, n);
    return new u(o, (a) => _r(a));
  }
  async transactionSubmit(t, n) {
    return await (await this.transactionSubmitRaw(t, n)).value();
  }
}
export {
  Tr as AtLedgerStateMixinFromJSON,
  Ve as AtLedgerStateMixinFromJSONTyped,
  wr as AtLedgerStateMixinToJSON,
  De as BASE_PATH,
  S as BaseAPI,
  Nr as BlobApiResponse,
  gr as COLLECTION_FORMATS,
  v as CommittedTransactionInfoFromJSON,
  Ge as CommittedTransactionInfoFromJSONTyped,
  x as CommittedTransactionInfoToJSON,
  Ae as Configuration,
  ve as DefaultConfig,
  Ar as EntityDetailsRequestFromJSON,
  Ue as EntityDetailsRequestFromJSONTyped,
  $e as EntityDetailsRequestToJSON,
  Yr as EntityDetailsResponseAllOfFromJSON,
  rt as EntityDetailsResponseAllOfFromJSONTyped,
  Zr as EntityDetailsResponseAllOfToJSON,
  kr as EntityDetailsResponseComponentDetailsFromJSON,
  K as EntityDetailsResponseComponentDetailsFromJSONTyped,
  ze as EntityDetailsResponseComponentDetailsToJSON,
  B as EntityDetailsResponseDetailsFromJSON,
  Ye as EntityDetailsResponseDetailsFromJSONTyped,
  z as EntityDetailsResponseDetailsToJSON,
  vr as EntityDetailsResponseDetailsType,
  h as EntityDetailsResponseDetailsTypeFromJSON,
  Be as EntityDetailsResponseDetailsTypeFromJSONTyped,
  xr as EntityDetailsResponseDetailsTypeToJSON,
  tt as EntityDetailsResponseFromJSON,
  nt as EntityDetailsResponseFromJSONTyped,
  Mr as EntityDetailsResponseFungibleResourceDetailsFromJSON,
  G as EntityDetailsResponseFungibleResourceDetailsFromJSONTyped,
  He as EntityDetailsResponseFungibleResourceDetailsToJSON,
  Gr as EntityDetailsResponseNonFungibleResourceDetailsFromJSON,
  U as EntityDetailsResponseNonFungibleResourceDetailsFromJSONTyped,
  Qe as EntityDetailsResponseNonFungibleResourceDetailsToJSON,
  $r as EntityDetailsResponsePackageDetailsFromJSON,
  $ as EntityDetailsResponsePackageDetailsFromJSONTyped,
  Xe as EntityDetailsResponsePackageDetailsToJSON,
  Qr as EntityDetailsResponseToJSON,
  ei as EntityFungiblesRequestFromJSON,
  it as EntityFungiblesRequestFromJSONTyped,
  st as EntityFungiblesRequestToJSON,
  oi as EntityFungiblesResponseAllOfFromJSON,
  ct as EntityFungiblesResponseAllOfFromJSONTyped,
  ai as EntityFungiblesResponseAllOfToJSON,
  ut as EntityFungiblesResponseFromJSON,
  dt as EntityFungiblesResponseFromJSONTyped,
  ii as EntityFungiblesResponseToJSON,
  di as EntityMetadataCollectionAllOfFromJSON,
  lt as EntityMetadataCollectionAllOfFromJSONTyped,
  ci as EntityMetadataCollectionAllOfToJSON,
  m as EntityMetadataCollectionFromJSON,
  je as EntityMetadataCollectionFromJSONTyped,
  p as EntityMetadataCollectionToJSON,
  H as EntityMetadataItemFromJSON,
  Ze as EntityMetadataItemFromJSONTyped,
  W as EntityMetadataItemToJSON,
  mi as EntityMetadataRequestAllOfFromJSON,
  mt as EntityMetadataRequestAllOfFromJSONTyped,
  pi as EntityMetadataRequestAllOfToJSON,
  fi as EntityMetadataRequestFromJSON,
  ft as EntityMetadataRequestFromJSONTyped,
  _t as EntityMetadataRequestToJSON,
  Ri as EntityMetadataResponseAllOfFromJSON,
  gt as EntityMetadataResponseAllOfFromJSONTyped,
  Si as EntityMetadataResponseAllOfToJSON,
  pt as EntityMetadataResponseFromJSON,
  yt as EntityMetadataResponseFromJSONTyped,
  gi as EntityMetadataResponseToJSON,
  bi as EntityNonFungibleIdsRequestAllOfFromJSON,
  St as EntityNonFungibleIdsRequestAllOfFromJSONTyped,
  Ti as EntityNonFungibleIdsRequestAllOfToJSON,
  hi as EntityNonFungibleIdsRequestFromJSON,
  Ot as EntityNonFungibleIdsRequestFromJSONTyped,
  Rt as EntityNonFungibleIdsRequestToJSON,
  Di as EntityNonFungibleIdsResponseAllOfFromJSON,
  Tt as EntityNonFungibleIdsResponseAllOfFromJSONTyped,
  Ai as EntityNonFungibleIdsResponseAllOfToJSON,
  Ft as EntityNonFungibleIdsResponseFromJSON,
  bt as EntityNonFungibleIdsResponseFromJSONTyped,
  Ei as EntityNonFungibleIdsResponseToJSON,
  xi as EntityNonFungiblesRequestFromJSON,
  wt as EntityNonFungiblesRequestFromJSONTyped,
  Jt as EntityNonFungiblesRequestToJSON,
  Li as EntityNonFungiblesResponseAllOfFromJSON,
  At as EntityNonFungiblesResponseAllOfFromJSONTyped,
  Ki as EntityNonFungiblesResponseAllOfToJSON,
  qt as EntityNonFungiblesResponseFromJSON,
  Dt as EntityNonFungiblesResponseFromJSONTyped,
  Mi as EntityNonFungiblesResponseToJSON,
  $i as EntityNotFoundErrorFromJSON,
  te as EntityNotFoundErrorFromJSONTyped,
  Bi as EntityNotFoundErrorToJSON,
  Qi as EntityOverviewRequestAllOfFromJSON,
  kt as EntityOverviewRequestAllOfFromJSONTyped,
  Xi as EntityOverviewRequestAllOfToJSON,
  Hi as EntityOverviewRequestFromJSON,
  xt as EntityOverviewRequestFromJSONTyped,
  Ct as EntityOverviewRequestToJSON,
  ts as EntityOverviewResponseAllOfFromJSON,
  Lt as EntityOverviewResponseAllOfFromJSONTyped,
  ns as EntityOverviewResponseAllOfToJSON,
  ne as EntityOverviewResponseEntityItemFromJSON,
  Pt as EntityOverviewResponseEntityItemFromJSONTyped,
  re as EntityOverviewResponseEntityItemToJSON,
  Mt as EntityOverviewResponseFromJSON,
  Vt as EntityOverviewResponseFromJSONTyped,
  ji as EntityOverviewResponseToJSON,
  os as EntityResourcesRequestAllOfFromJSON,
  Ut as EntityResourcesRequestAllOfFromJSONTyped,
  as as EntityResourcesRequestAllOfToJSON,
  is as EntityResourcesRequestFromJSON,
  Kt as EntityResourcesRequestFromJSONTyped,
  Gt as EntityResourcesRequestToJSON,
  ls as EntityResourcesResponseAllOfFromJSON,
  zt as EntityResourcesResponseAllOfFromJSONTyped,
  fs as EntityResourcesResponseAllOfToJSON,
  $t as EntityResourcesResponseFromJSON,
  Bt as EntityResourcesResponseFromJSONTyped,
  ds as EntityResourcesResponseToJSON,
  ms as ErrorResponseFromJSON,
  Ht as ErrorResponseFromJSONTyped,
  ps as ErrorResponseToJSON,
  Pe as FetchError,
  gs as FungibleResourcesCollectionAllOfFromJSON,
  Wt as FungibleResourcesCollectionAllOfFromJSONTyped,
  Os as FungibleResourcesCollectionAllOfToJSON,
  F as FungibleResourcesCollectionFromJSON,
  at as FungibleResourcesCollectionFromJSONTyped,
  Q as FungibleResourcesCollectionItemFromJSON,
  ot as FungibleResourcesCollectionItemFromJSONTyped,
  X as FungibleResourcesCollectionItemToJSON,
  b as FungibleResourcesCollectionToJSON,
  vt as GatewayErrorFromJSON,
  y as GatewayErrorFromJSONTyped,
  g as GatewayErrorToJSON,
  ie as GatewayInfoResponseKnownTargetFromJSON,
  Qt as GatewayInfoResponseKnownTargetFromJSONTyped,
  se as GatewayInfoResponseKnownTargetToJSON,
  oe as GatewayInfoResponseReleaseInfoFromJSON,
  Xt as GatewayInfoResponseReleaseInfoFromJSONTyped,
  ae as GatewayInfoResponseReleaseInfoToJSON,
  Ts as GatewayInformationResponseAllOfFromJSON,
  en as GatewayInformationResponseAllOfFromJSONTyped,
  ws as GatewayInformationResponseAllOfToJSON,
  ue as GatewayInformationResponseAllOfWellKnownAddressesFromJSON,
  Yt as GatewayInformationResponseAllOfWellKnownAddressesFromJSONTyped,
  de as GatewayInformationResponseAllOfWellKnownAddressesToJSON,
  Zt as GatewayInformationResponseFromJSON,
  jt as GatewayInformationResponseFromJSONTyped,
  Fs as GatewayInformationResponseToJSON,
  Ds as InternalServerErrorAllOfFromJSON,
  tn as InternalServerErrorAllOfFromJSONTyped,
  As as InternalServerErrorAllOfToJSON,
  Is as InternalServerErrorFromJSON,
  ce as InternalServerErrorFromJSONTyped,
  Es as InternalServerErrorToJSON,
  xs as InvalidEntityErrorFromJSON,
  le as InvalidEntityErrorFromJSONTyped,
  Cs as InvalidEntityErrorToJSON,
  Ks as InvalidRequestErrorAllOfFromJSON,
  rn as InvalidRequestErrorAllOfFromJSONTyped,
  Gs as InvalidRequestErrorAllOfToJSON,
  Ms as InvalidRequestErrorFromJSON,
  me as InvalidRequestErrorFromJSONTyped,
  Vs as InvalidRequestErrorToJSON,
  $s as InvalidTransactionErrorFromJSON,
  pe as InvalidTransactionErrorFromJSONTyped,
  Bs as InvalidTransactionErrorToJSON,
  u as JSONApiResponse,
  d as LedgerStateFromJSON,
  et as LedgerStateFromJSONTyped,
  Hs as LedgerStateMixinFromJSON,
  sn as LedgerStateMixinFromJSONTyped,
  Ws as LedgerStateMixinToJSON,
  f as LedgerStateSelectorFromJSON,
  Me as LedgerStateSelectorFromJSONTyped,
  _ as LedgerStateSelectorToJSON,
  c as LedgerStateToJSON,
  Zs as NonFungibleDataRequestAllOfFromJSON,
  un as NonFungibleDataRequestAllOfFromJSONTyped,
  js as NonFungibleDataRequestAllOfToJSON,
  Xs as NonFungibleDataRequestFromJSON,
  on as NonFungibleDataRequestFromJSONTyped,
  an as NonFungibleDataRequestToJSON,
  ro as NonFungibleDataResponseAllOfFromJSON,
  ln as NonFungibleDataResponseAllOfFromJSONTyped,
  io as NonFungibleDataResponseAllOfToJSON,
  dn as NonFungibleDataResponseFromJSON,
  cn as NonFungibleDataResponseFromJSONTyped,
  to as NonFungibleDataResponseToJSON,
  Vr as NonFungibleIdType,
  C as NonFungibleIdTypeFromJSON,
  We as NonFungibleIdTypeFromJSONTyped,
  Lr as NonFungibleIdTypeToJSON,
  oo as NonFungibleIdsCollectionAllOfFromJSON,
  fn as NonFungibleIdsCollectionAllOfFromJSONTyped,
  ao as NonFungibleIdsCollectionAllOfToJSON,
  T as NonFungibleIdsCollectionFromJSON,
  ht as NonFungibleIdsCollectionFromJSONTyped,
  Y as NonFungibleIdsCollectionItemFromJSON,
  Nt as NonFungibleIdsCollectionItemFromJSONTyped,
  Z as NonFungibleIdsCollectionItemToJSON,
  w as NonFungibleIdsCollectionToJSON,
  co as NonFungibleIdsRequestFromJSON,
  _n as NonFungibleIdsRequestFromJSONTyped,
  mn as NonFungibleIdsRequestToJSON,
  mo as NonFungibleIdsResponseAllOfFromJSON,
  gn as NonFungibleIdsResponseAllOfFromJSONTyped,
  po as NonFungibleIdsResponseAllOfToJSON,
  pn as NonFungibleIdsResponseFromJSON,
  yn as NonFungibleIdsResponseFromJSONTyped,
  fo as NonFungibleIdsResponseToJSON,
  go as NonFungibleResourcesCollectionAllOfFromJSON,
  On as NonFungibleResourcesCollectionAllOfFromJSONTyped,
  Oo as NonFungibleResourcesCollectionAllOfToJSON,
  J as NonFungibleResourcesCollectionFromJSON,
  Et as NonFungibleResourcesCollectionFromJSONTyped,
  j as NonFungibleResourcesCollectionItemFromJSON,
  It as NonFungibleResourcesCollectionItemFromJSONTyped,
  ee as NonFungibleResourcesCollectionItemToJSON,
  I as NonFungibleResourcesCollectionToJSON,
  Fo as NotSyncedUpErrorAllOfFromJSON,
  Rn as NotSyncedUpErrorAllOfFromJSONTyped,
  bo as NotSyncedUpErrorAllOfToJSON,
  So as NotSyncedUpErrorFromJSON,
  ye as NotSyncedUpErrorFromJSONTyped,
  No as NotSyncedUpErrorToJSON,
  Ao as PublicKeyEcdsaSecp256k1AllOfFromJSON,
  Fn as PublicKeyEcdsaSecp256k1AllOfFromJSONTyped,
  vo as PublicKeyEcdsaSecp256k1AllOfToJSON,
  Eo as PublicKeyEcdsaSecp256k1FromJSON,
  ge as PublicKeyEcdsaSecp256k1FromJSONTyped,
  qo as PublicKeyEcdsaSecp256k1ToJSON,
  Mo as PublicKeyEddsaEd25519AllOfFromJSON,
  bn as PublicKeyEddsaEd25519AllOfFromJSONTyped,
  Vo as PublicKeyEddsaEd25519AllOfToJSON,
  Co as PublicKeyEddsaEd25519FromJSON,
  Oe as PublicKeyEddsaEd25519FromJSONTyped,
  ko as PublicKeyEddsaEd25519ToJSON,
  hn as PublicKeyFromJSON,
  k as PublicKeyFromJSONTyped,
  P as PublicKeyToJSON,
  To as PublicKeyType,
  Sn as PublicKeyTypeFromJSON,
  Nn as PublicKeyTypeFromJSONTyped,
  wo as PublicKeyTypeToJSON,
  l as RequiredError,
  ke as ResponseError,
  Ko as ResultSetCursorMixinFromJSON,
  Tn as ResultSetCursorMixinFromJSONTyped,
  Go as ResultSetCursorMixinToJSON,
  Xa as StateApi,
  zo as StateValidatorsListRequestAllOfFromJSON,
  In as StateValidatorsListRequestAllOfFromJSONTyped,
  Ho as StateValidatorsListRequestAllOfToJSON,
  $o as StateValidatorsListRequestFromJSON,
  wn as StateValidatorsListRequestFromJSONTyped,
  Jn as StateValidatorsListRequestToJSON,
  ta as StateValidatorsListResponseAllOfFromJSON,
  Vn as StateValidatorsListResponseAllOfFromJSONTyped,
  na as StateValidatorsListResponseAllOfToJSON,
  Pn as StateValidatorsListResponseFromJSON,
  Mn as StateValidatorsListResponseFromJSONTyped,
  jo as StateValidatorsListResponseToJSON,
  Ya as StatusApi,
  Za as StreamApi,
  hr as TextApiResponse,
  D as TokenAmountFromJSON,
  Le as TokenAmountFromJSONTyped,
  A as TokenAmountToJSON,
  ja as TransactionApi,
  da as TransactionCommittedDetailsRequestAllOfFromJSON,
  Bn as TransactionCommittedDetailsRequestAllOfFromJSONTyped,
  ca as TransactionCommittedDetailsRequestAllOfToJSON,
  aa as TransactionCommittedDetailsRequestFromJSON,
  Un as TransactionCommittedDetailsRequestFromJSONTyped,
  E as TransactionCommittedDetailsRequestIdentifierFromJSON,
  Gn as TransactionCommittedDetailsRequestIdentifierFromJSONTyped,
  q as TransactionCommittedDetailsRequestIdentifierToJSON,
  ra as TransactionCommittedDetailsRequestIdentifierType,
  Ln as TransactionCommittedDetailsRequestIdentifierTypeFromJSON,
  Kn as TransactionCommittedDetailsRequestIdentifierTypeFromJSONTyped,
  ia as TransactionCommittedDetailsRequestIdentifierTypeToJSON,
  $n as TransactionCommittedDetailsRequestToJSON,
  pa as TransactionCommittedDetailsResponseAllOfFromJSON,
  Qn as TransactionCommittedDetailsResponseAllOfFromJSONTyped,
  ya as TransactionCommittedDetailsResponseAllOfToJSON,
  Fe as TransactionCommittedDetailsResponseDetailsFromJSON,
  zn as TransactionCommittedDetailsResponseDetailsFromJSONTyped,
  be as TransactionCommittedDetailsResponseDetailsToJSON,
  Hn as TransactionCommittedDetailsResponseFromJSON,
  Wn as TransactionCommittedDetailsResponseFromJSONTyped,
  _a as TransactionCommittedDetailsResponseToJSON,
  Xn as TransactionConstructionResponseFromJSON,
  Yn as TransactionConstructionResponseFromJSONTyped,
  Oa as TransactionConstructionResponseToJSON,
  Fa as TransactionNotFoundErrorAllOfFromJSON,
  Zn as TransactionNotFoundErrorAllOfFromJSONTyped,
  ba as TransactionNotFoundErrorAllOfToJSON,
  Sa as TransactionNotFoundErrorFromJSON,
  Te as TransactionNotFoundErrorFromJSONTyped,
  Na as TransactionNotFoundErrorToJSON,
  Ia as TransactionRecentRequestAllOfFromJSON,
  tr as TransactionRecentRequestAllOfFromJSONTyped,
  Ea as TransactionRecentRequestAllOfToJSON,
  wa as TransactionRecentRequestFromJSON,
  jn as TransactionRecentRequestFromJSONTyped,
  er as TransactionRecentRequestToJSON,
  nr as TransactionRecentResponseFromJSON,
  rr as TransactionRecentResponseFromJSONTyped,
  Da as TransactionRecentResponseToJSON,
  Ir as TransactionStatus,
  N as TransactionStatusFromJSON,
  Ke as TransactionStatusFromJSONTyped,
  Ca as TransactionStatusRequestAllOfFromJSON,
  or as TransactionStatusRequestAllOfFromJSONTyped,
  ka as TransactionStatusRequestAllOfToJSON,
  va as TransactionStatusRequestFromJSON,
  ir as TransactionStatusRequestFromJSONTyped,
  sr as TransactionStatusRequestToJSON,
  Ka as TransactionStatusResponseAllOfFromJSON,
  cr as TransactionStatusResponseAllOfFromJSONTyped,
  Ga as TransactionStatusResponseAllOfToJSON,
  ur as TransactionStatusResponseFromJSON,
  dr as TransactionStatusResponseFromJSONTyped,
  we as TransactionStatusResponseKnownPayloadItemFromJSON,
  ar as TransactionStatusResponseKnownPayloadItemFromJSONTyped,
  Je as TransactionStatusResponseKnownPayloadItemToJSON,
  Va as TransactionStatusResponseToJSON,
  Er as TransactionStatusToJSON,
  $a as TransactionSubmitRequestFromJSON,
  lr as TransactionSubmitRequestFromJSONTyped,
  fr as TransactionSubmitRequestToJSON,
  _r as TransactionSubmitResponseFromJSON,
  mr as TransactionSubmitResponseFromJSONTyped,
  za as TransactionSubmitResponseToJSON,
  fe as ValidationErrorsAtPathFromJSON,
  nn as ValidationErrorsAtPathFromJSONTyped,
  _e as ValidationErrorsAtPathToJSON,
  Wa as ValidatorCollectionAllOfFromJSON,
  pr as ValidatorCollectionAllOfFromJSONTyped,
  Qa as ValidatorCollectionAllOfToJSON,
  Ne as ValidatorCollectionFromJSON,
  kn as ValidatorCollectionFromJSONTyped,
  An as ValidatorCollectionItemActiveInEpochFromJSON,
  vn as ValidatorCollectionItemActiveInEpochFromJSONTyped,
  En as ValidatorCollectionItemActiveInEpochStakeFromJSON,
  qn as ValidatorCollectionItemActiveInEpochStakeFromJSONTyped,
  Dn as ValidatorCollectionItemActiveInEpochStakeToJSON,
  xn as ValidatorCollectionItemActiveInEpochToJSON,
  Re as ValidatorCollectionItemFromJSON,
  Cn as ValidatorCollectionItemFromJSONTyped,
  Se as ValidatorCollectionItemToJSON,
  he as ValidatorCollectionToJSON,
  Sr as VoidApiResponse,
  Rr as canConsumeForm,
  i as exists,
  br as instanceOfAtLedgerStateMixin,
  qr as instanceOfCommittedTransactionInfo,
  Dr as instanceOfEntityDetailsRequest,
  Wr as instanceOfEntityDetailsResponse,
  Xr as instanceOfEntityDetailsResponseAllOf,
  Cr as instanceOfEntityDetailsResponseComponentDetails,
  Pr as instanceOfEntityDetailsResponseFungibleResourceDetails,
  Kr as instanceOfEntityDetailsResponseNonFungibleResourceDetails,
  Ur as instanceOfEntityDetailsResponsePackageDetails,
  jr as instanceOfEntityFungiblesRequest,
  ri as instanceOfEntityFungiblesResponse,
  si as instanceOfEntityFungiblesResponseAllOf,
  zr as instanceOfEntityMetadataCollection,
  ui as instanceOfEntityMetadataCollectionAllOf,
  Br as instanceOfEntityMetadataItem,
  li as instanceOfEntityMetadataRequest,
  _i as instanceOfEntityMetadataRequestAllOf,
  yi as instanceOfEntityMetadataResponse,
  Oi as instanceOfEntityMetadataResponseAllOf,
  Ni as instanceOfEntityNonFungibleIdsRequest,
  Fi as instanceOfEntityNonFungibleIdsRequestAllOf,
  Ii as instanceOfEntityNonFungibleIdsResponse,
  qi as instanceOfEntityNonFungibleIdsResponseAllOf,
  vi as instanceOfEntityNonFungiblesRequest,
  Pi as instanceOfEntityNonFungiblesResponse,
  Vi as instanceOfEntityNonFungiblesResponseAllOf,
  Ui as instanceOfEntityNotFoundError,
  zi as instanceOfEntityOverviewRequest,
  Wi as instanceOfEntityOverviewRequestAllOf,
  Zi as instanceOfEntityOverviewResponse,
  es as instanceOfEntityOverviewResponseAllOf,
  Yi as instanceOfEntityOverviewResponseEntityItem,
  rs as instanceOfEntityResourcesRequest,
  ss as instanceOfEntityResourcesRequestAllOf,
  us as instanceOfEntityResourcesResponse,
  cs as instanceOfEntityResourcesResponseAllOf,
  _s as instanceOfErrorResponse,
  ni as instanceOfFungibleResourcesCollection,
  ys as instanceOfFungibleResourcesCollectionAllOf,
  ti as instanceOfFungibleResourcesCollectionItem,
  Gi as instanceOfGatewayError,
  Rs as instanceOfGatewayInfoResponseKnownTarget,
  Ss as instanceOfGatewayInfoResponseReleaseInfo,
  hs as instanceOfGatewayInformationResponse,
  bs as instanceOfGatewayInformationResponseAllOf,
  Ns as instanceOfGatewayInformationResponseAllOfWellKnownAddresses,
  Js as instanceOfInternalServerError,
  qs as instanceOfInternalServerErrorAllOf,
  vs as instanceOfInvalidEntityError,
  Ps as instanceOfInvalidRequestError,
  Ls as instanceOfInvalidRequestErrorAllOf,
  Us as instanceOfInvalidTransactionError,
  Hr as instanceOfLedgerState,
  zs as instanceOfLedgerStateMixin,
  Fr as instanceOfLedgerStateSelector,
  Qs as instanceOfNonFungibleDataRequest,
  Ys as instanceOfNonFungibleDataRequestAllOf,
  eo as instanceOfNonFungibleDataResponse,
  no as instanceOfNonFungibleDataResponseAllOf,
  Ji as instanceOfNonFungibleIdsCollection,
  so as instanceOfNonFungibleIdsCollectionAllOf,
  wi as instanceOfNonFungibleIdsCollectionItem,
  uo as instanceOfNonFungibleIdsRequest,
  lo as instanceOfNonFungibleIdsResponse,
  _o as instanceOfNonFungibleIdsResponseAllOf,
  ki as instanceOfNonFungibleResourcesCollection,
  yo as instanceOfNonFungibleResourcesCollectionAllOf,
  Ci as instanceOfNonFungibleResourcesCollectionItem,
  Ro as instanceOfNotSyncedUpError,
  ho as instanceOfNotSyncedUpErrorAllOf,
  Jo as instanceOfPublicKey,
  Io as instanceOfPublicKeyEcdsaSecp256k1,
  Do as instanceOfPublicKeyEcdsaSecp256k1AllOf,
  xo as instanceOfPublicKeyEddsaEd25519,
  Po as instanceOfPublicKeyEddsaEd25519AllOf,
  Lo as instanceOfResultSetCursorMixin,
  Uo as instanceOfStateValidatorsListRequest,
  Bo as instanceOfStateValidatorsListRequestAllOf,
  Zo as instanceOfStateValidatorsListResponse,
  ea as instanceOfStateValidatorsListResponseAllOf,
  Jr as instanceOfTokenAmount,
  oa as instanceOfTransactionCommittedDetailsRequest,
  ua as instanceOfTransactionCommittedDetailsRequestAllOf,
  sa as instanceOfTransactionCommittedDetailsRequestIdentifier,
  fa as instanceOfTransactionCommittedDetailsResponse,
  ma as instanceOfTransactionCommittedDetailsResponseAllOf,
  la as instanceOfTransactionCommittedDetailsResponseDetails,
  ga as instanceOfTransactionConstructionResponse,
  Ra as instanceOfTransactionNotFoundError,
  ha as instanceOfTransactionNotFoundErrorAllOf,
  Ta as instanceOfTransactionRecentRequest,
  Ja as instanceOfTransactionRecentRequestAllOf,
  qa as instanceOfTransactionRecentResponse,
  Aa as instanceOfTransactionStatusRequest,
  xa as instanceOfTransactionStatusRequestAllOf,
  Ma as instanceOfTransactionStatusResponse,
  La as instanceOfTransactionStatusResponseAllOf,
  Pa as instanceOfTransactionStatusResponseKnownPayloadItem,
  Ua as instanceOfTransactionSubmitRequest,
  Ba as instanceOfTransactionSubmitResponse,
  ks as instanceOfValidationErrorsAtPath,
  Yo as instanceOfValidatorCollection,
  Ha as instanceOfValidatorCollectionAllOf,
  Xo as instanceOfValidatorCollectionItem,
  Qo as instanceOfValidatorCollectionItemActiveInEpoch,
  Wo as instanceOfValidatorCollectionItemActiveInEpochStake,
  Or as mapValues,
  V as querystring
};
